<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Testing pleiotropy vs. separate QTL in multiparental populations • qtl2pleio</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Testing pleiotropy vs. separate QTL in multiparental populations">
<meta property="og:description" content="The package contains code that implements an
    adaptation of Jiang &amp; Zeng's (1995) likelihood ratio test for testing
    the null hypothesis of pleiotropy against the alternative hypothesis,
    two separate QTL. The test differs from that in Jiang &amp; Zeng (1995) in
    that our test accommodates multiparental populations, such as the Diversity 
    Outbred mouse population.">
<meta property="og:image" content="https://fboehm.us/static/software/qtl2pleio/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">qtl2pleio</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.2.9001</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/bootstrap-cluster.html">Using HTCondor for bootstrap analysis</a>
    </li>
    <li>
      <a href="articles/recla-analysis.html">Recla Analysis</a>
    </li>
    <li>
      <a href="articles/testing-pleiotropy-vs-separate-qtl.html">Testing pleiotropy v separate QTL in multiparental populations</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/fboehm/qtl2pleio">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    

    
    
<!-- README.md is generated from README.Rmd. Please edit README.Rmd -->
<div id="qtl2pleio" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#qtl2pleio" class="anchor"></a>qtl2pleio <img src="https://raw.githubusercontent.com/fboehm/qtl2pleio/master/man/figures/logo.png" align="right">
</h1></div>

<div id="goals" class="section level2">
<h2 class="hasAnchor">
<a href="#goals" class="anchor"></a>Goals</h2>
<p>The goal of qtl2pleio is, for a pair of traits that show evidence for a QTL in a common region, to distinguish between pleiotropy (the null hypothesis, that they are affected by a common QTL) and the alternative that they are affected by separate QTL. It extends the likelihood ratio test of <a href="http://www.genetics.org/content/140/3/1111.long">Jiang and Zeng (1995)</a> for multiparental populations, such as Diversity Outbred mice, including the use of a linear mixed model to account for population structure. <code>qtl2pleio</code> data structures are those used in the <a href="https://kbroman.org/qtl2"><code>rqtl/qtl2</code></a> package.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To install qtl2pleio, use <code>install_github()</code> from the <a href="https://devtools.r-lib.org">devtools</a> package.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/install.packages">install.packages</a></span>(<span class="st">"devtools"</span>)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"fboehm/qtl2pleio"</span>)</a></code></pre></div>
<p>You may also wish to install <a href="https://kbroman.org/qtl2">R/qtl2</a> and the <a href="https://github.com/rqtl/qtl2convert"><code>qtl2convert</code></a> package. We will use both below.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/install.packages">install.packages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"qtl2"</span>, <span class="st">"qtl2convert"</span>), <span class="dt">repos=</span><span class="st">"http://rqtl.org/qtl2"</span>)</a></code></pre></div>
</div>
<div id="example" class="section level2">
<h2 class="hasAnchor">
<a href="#example" class="anchor"></a>Example</h2>
<p>Below, we walk through an example analysis with Diversity Outbred mouse data. We need a number of preliminary steps before we can perform our test of pleiotropy vs. separate QTL. Many procedures rely on the R package <code>qtl2</code>. We first load the <code>qtl2</code>, <code>qtl2convert</code>, and <code>qtl2pleio</code> packages.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(qtl2)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(qtl2convert)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(qtl2pleio)</a></code></pre></div>
<div id="reading-data-from-qtl2data-repository-on-github" class="section level3">
<h3 class="hasAnchor">
<a href="#reading-data-from-qtl2data-repository-on-github" class="anchor"></a>Reading data from <code>qtl2data</code> repository on github</h3>
<p>We’ll consider the <a href="https://github.com/rqtl/qtl2data/tree/master/DOex"><code>DOex</code></a> data in the <a href="https://github.com/rqtl/qtl2data"><code>qtl2data</code></a> repository. We’ll download the pre-calculated allele probabilities.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">tmpfile &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/tempfile">tempfile</a></span>()</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">file &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(<span class="st">"https://raw.githubusercontent.com/rqtl/"</span>,</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">               <span class="st">"qtl2data/master/DOex/DOex_alleleprobs.rds"</span>)</a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/download.file">download.file</a></span>(file, tmpfile)</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">pr &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/readRDS">readRDS</a></span>(tmpfile)</a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlink">unlink</a></span>(tmpfile)</a></code></pre></div>
<p>And then we’ll download the physical marker map and convert it to a R/qtl2 map object.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">tmpfile &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/tempfile">tempfile</a></span>()</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">file &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(<span class="st">"https://raw.githubusercontent.com/rqtl/"</span>,</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">               <span class="st">"qtl2data/master/DOex/DOex_pmap.csv"</span>)</a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/download.file">download.file</a></span>(file, tmpfile)</a>
<a class="sourceLine" id="cb5-5" data-line-number="5">pmap_pre &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/read.table">read.csv</a></span>(tmpfile)</a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlink">unlink</a></span>(tmpfile)</a>
<a class="sourceLine" id="cb5-7" data-line-number="7">pm &lt;-<span class="st"> </span>qtl2convert<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/qtl2convert/topics/map_df_to_list">map_df_to_list</a></span>(pmap_pre, <span class="dt">pos_column=</span><span class="st">"pos"</span>)</a></code></pre></div>
<p>We now have an allele probabilities object stored in <code>pr</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(pr)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="co">#&gt; [1] "2" "3" "X"</span></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/dim">dim</a></span>(pr<span class="op">$</span><span class="st">`</span><span class="dt">2</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co">#&gt; [1] 261   8 127</span></a></code></pre></div>
<p>We see that <code>pr</code> is a list of 3 three-dimensional arrays - one array for each of 3 chromosomes.</p>
</div>
<div id="kinship-calculations" class="section level3">
<h3 class="hasAnchor">
<a href="#kinship-calculations" class="anchor"></a>Kinship calculations</h3>
<p>For our statistical model, we need a kinship matrix. Although we don’t have genome-wide data - since we have allele probabilities for only 3 chromosomes - let’s calculate a kinship matrix using “leave-one-chromosome-out”. In practice, one would want to use allele probabilities from a full genome-wide set of markers.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">kinship &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/qtl2/topics/calc_kinship">calc_kinship</a></span>(<span class="dt">probs =</span> pr, <span class="dt">type =</span> <span class="st">"loco"</span>)</a></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/str">str</a></span>(kinship)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="co">#&gt; List of 3</span></a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="co">#&gt;  $ 2: num [1:261, 1:261] 0.6919 0.0707 0.2355 0.0558 0.0512 ...</span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="co">#&gt;   ..- attr(*, "n_pos")= int 195</span></a>
<a class="sourceLine" id="cb8-5" data-line-number="5"><span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span></a>
<a class="sourceLine" id="cb8-6" data-line-number="6"><span class="co">#&gt;   .. ..$ : chr [1:261] "1" "4" "5" "6" ...</span></a>
<a class="sourceLine" id="cb8-7" data-line-number="7"><span class="co">#&gt;   .. ..$ : chr [1:261] "1" "4" "5" "6" ...</span></a>
<a class="sourceLine" id="cb8-8" data-line-number="8"><span class="co">#&gt;  $ 3: num [1:261, 1:261] 0.6642 0.0638 0.2035 0.1128 0.0772 ...</span></a>
<a class="sourceLine" id="cb8-9" data-line-number="9"><span class="co">#&gt;   ..- attr(*, "n_pos")= int 220</span></a>
<a class="sourceLine" id="cb8-10" data-line-number="10"><span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span></a>
<a class="sourceLine" id="cb8-11" data-line-number="11"><span class="co">#&gt;   .. ..$ : chr [1:261] "1" "4" "5" "6" ...</span></a>
<a class="sourceLine" id="cb8-12" data-line-number="12"><span class="co">#&gt;   .. ..$ : chr [1:261] "1" "4" "5" "6" ...</span></a>
<a class="sourceLine" id="cb8-13" data-line-number="13"><span class="co">#&gt;  $ X: num [1:261, 1:261] 0.4853 0.0821 0.1958 0.1039 0.1123 ...</span></a>
<a class="sourceLine" id="cb8-14" data-line-number="14"><span class="co">#&gt;   ..- attr(*, "n_pos")= int 229</span></a>
<a class="sourceLine" id="cb8-15" data-line-number="15"><span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span></a>
<a class="sourceLine" id="cb8-16" data-line-number="16"><span class="co">#&gt;   .. ..$ : chr [1:261] "1" "4" "5" "6" ...</span></a>
<a class="sourceLine" id="cb8-17" data-line-number="17"><span class="co">#&gt;   .. ..$ : chr [1:261] "1" "4" "5" "6" ...</span></a></code></pre></div>
<p>We see that <code>kinship</code> is a list containing 3 matrices. Each matrix is 261 by 261 - where the number of subjects is 261 - and symmetric. The <span class="math inline">\((i, j)\)</span> cell in the matrix contains the estimate of identity-by-state (IBS) probability for randomly chosen alleles at a single locus for those two subjects.</p>
<p>Before we simulate phenotype data, we first specify our statistical model.</p>
<p>We use the model:</p>
<p><span class="math display">\[vec(Y) = X vec(B) + vec(G) + vec(E)\]</span></p>
<p>where <span class="math inline">\(Y\)</span> is a <span class="math inline">\(n\)</span> by <span class="math inline">\(2\)</span> matrix, where each row is one subject and each column is one quantitative trait. <span class="math inline">\(X\)</span> is a <span class="math inline">\(2n\)</span> by <span class="math inline">\(2f\)</span> design matrix containing <span class="math inline">\(n\)</span> by <span class="math inline">\(f\)</span> allele probabilities matrices for each of two (possibly identical) markers. Thus, <span class="math inline">\(X\)</span> is a block-diagonal matrix, with exactly two <span class="math inline">\(n\)</span> by <span class="math inline">\(f\)</span> blocks on the diagonal. <span class="math inline">\(B\)</span> is a <span class="math inline">\(f\)</span> by 2 matrix. “vec” refers to the vectorization operator. “vec(B)”, where <span class="math inline">\(B\)</span> is a <span class="math inline">\(f\)</span> by <span class="math inline">\(2\)</span> matrix, is, thus, a (column) vector of length <span class="math inline">\(2f\)</span> that is formed by stacking the second column of <span class="math inline">\(B\)</span> beneath the first column of <span class="math inline">\(B\)</span>.</p>
<p><span class="math inline">\(G\)</span> is a matrix of random effects. We specify its distribution as matrix-variate normal with mean being a <span class="math inline">\(n\)</span> by <span class="math inline">\(2\)</span> matrix of zeros, covariance among row vectors a <span class="math inline">\(n\)</span> by <span class="math inline">\(n\)</span> kinship matrix, <span class="math inline">\(K\)</span>, and covariance among column vectors a <span class="math inline">\(2\)</span> by <span class="math inline">\(2\)</span> genetic covariance matrix, <span class="math inline">\(V_g\)</span>.</p>
<p>In mathematical notation, we write:</p>
<p><span class="math display">\[G \sim MN_{\text{n by 2}}(0, K, V_g)\]</span></p>
<p>We also need to specify the distribution of the <span class="math inline">\(E\)</span> matrix, which contains the random errors. <span class="math inline">\(E\)</span> is a random <span class="math inline">\(n\)</span> by <span class="math inline">\(2\)</span> matrix that is distributed as a matrix-variate normal distribution with mean being the <span class="math inline">\(n\)</span> by <span class="math inline">\(2\)</span> zero matrix, covariance among row vectors <span class="math inline">\(I_n\)</span>, the <span class="math inline">\(n\)</span> by <span class="math inline">\(n\)</span> identity matrix, and covariance among columns the <span class="math inline">\(2\)</span> by <span class="math inline">\(2\)</span> matrix <span class="math inline">\(V_e\)</span>.</p>
<p><span class="math display">\[E \sim MN_{\text{n by 2}}(0, I_n, V_e)\]</span></p>
<p>In practice, we typically observe the phenotype matrix <span class="math inline">\(Y\)</span>. We also treat as known the design matrix <span class="math inline">\(X\)</span> and the kinship matrix <span class="math inline">\(K\)</span>. We then infer the values of <span class="math inline">\(B\)</span>, <span class="math inline">\(V_g\)</span>, and <span class="math inline">\(V_e\)</span>.</p>
</div>
<div id="simulating-phenotypes-with-qtl2pleiosim1" class="section level3">
<h3 class="hasAnchor">
<a href="#simulating-phenotypes-with-qtl2pleiosim1" class="anchor"></a>Simulating phenotypes with <code><a href="reference/sim1.html">qtl2pleio::sim1</a></code>
</h3>
<p>The function to simulate phenotypes in <code>qtl2pleio</code> is <code>sim1</code>. By examining its help page, we see that it takes five arguments. The help page also gives the dimensions of the inputs.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="co"># set up the design matrix, X</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2">pp &lt;-<span class="st"> </span>pr[[<span class="dv">2</span>]] <span class="co">#we'll with Chr 3's genotype data</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/dim">dim</a></span>(pp)</a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="co">#&gt; [1] 261   8 102</span></a></code></pre></div>
<p>We prepare a block-diagonal design matrix X that contains two nonzero blocks on the diagonal, one for each trait. We use here a function from the <code>gemma2</code> R package to set up the needed matrix.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="co">#Next, we prepare a design matrix X</span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2">X &lt;-<span class="st"> </span>gemma2<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/gemma2/topics/stagger_mats">stagger_mats</a></span>(pp[ , , <span class="dv">50</span>], pp[ , , <span class="dv">50</span>])</a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/dim">dim</a></span>(X)</a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="co">#&gt; [1] 522  16</span></a></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="co"># assemble B matrix of allele effects</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2">B &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">matrix</a></span>(<span class="dt">data =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="op">-</span><span class="dv">1</span>, <span class="dv">-1</span>, <span class="dv">-1</span>, <span class="dv">-1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">-1</span>, <span class="dv">-1</span>, <span class="dv">-1</span>, <span class="dv">-1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>), <span class="dt">nrow =</span> <span class="dv">8</span>, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">byrow =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="co"># verify that B is what we want:</span></a>
<a class="sourceLine" id="cb11-4" data-line-number="4">B</a>
<a class="sourceLine" id="cb11-5" data-line-number="5"><span class="co">#&gt;      [,1] [,2]</span></a>
<a class="sourceLine" id="cb11-6" data-line-number="6"><span class="co">#&gt; [1,]   -1   -1</span></a>
<a class="sourceLine" id="cb11-7" data-line-number="7"><span class="co">#&gt; [2,]   -1   -1</span></a>
<a class="sourceLine" id="cb11-8" data-line-number="8"><span class="co">#&gt; [3,]   -1   -1</span></a>
<a class="sourceLine" id="cb11-9" data-line-number="9"><span class="co">#&gt; [4,]   -1   -1</span></a>
<a class="sourceLine" id="cb11-10" data-line-number="10"><span class="co">#&gt; [5,]    1    1</span></a>
<a class="sourceLine" id="cb11-11" data-line-number="11"><span class="co">#&gt; [6,]    1    1</span></a>
<a class="sourceLine" id="cb11-12" data-line-number="12"><span class="co">#&gt; [7,]    1    1</span></a>
<a class="sourceLine" id="cb11-13" data-line-number="13"><span class="co">#&gt; [8,]    1    1</span></a>
<a class="sourceLine" id="cb11-14" data-line-number="14"><span class="co"># set.seed to ensure reproducibility</span></a>
<a class="sourceLine" id="cb11-15" data-line-number="15"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">2018-01-30</span>)</a>
<a class="sourceLine" id="cb11-16" data-line-number="16"><span class="co"># call to sim1</span></a>
<a class="sourceLine" id="cb11-17" data-line-number="17">Ypre &lt;-<span class="st"> </span><span class="kw"><a href="reference/sim1.html">sim1</a></span>(<span class="dt">X =</span> X, <span class="dt">B =</span> B, <span class="dt">Vg =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/diag">diag</a></span>(<span class="dv">2</span>), <span class="dt">Ve =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/diag">diag</a></span>(<span class="dv">2</span>), <span class="dt">kinship =</span> kinship[[<span class="dv">2</span>]])</a>
<a class="sourceLine" id="cb11-18" data-line-number="18">Y &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">matrix</a></span>(Ypre, <span class="dt">nrow =</span> <span class="dv">261</span>, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">byrow =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb11-19" data-line-number="19"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(Y) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(pp)</a>
<a class="sourceLine" id="cb11-20" data-line-number="20"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(Y) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"tr1"</span>, <span class="st">"tr2"</span>)</a></code></pre></div>
<p>Let’s perform univariate QTL mapping for each of the two traits in the Y matrix.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">s1 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/qtl2/topics/scan1">scan1</a></span>(<span class="dt">genoprobs =</span> pr, <span class="dt">pheno =</span> Y, <span class="dt">kinship =</span> kinship)</a></code></pre></div>
<p>Here is a plot of the results.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(s1, pm)</a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(s1, pm, <span class="dt">lod=</span><span class="dv">2</span>, <span class="dt">col=</span><span class="st">"violetred"</span>, <span class="dt">add=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb13-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/legend">legend</a></span>(<span class="st">"topleft"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(s1), <span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">col=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"darkslateblue"</span>, <span class="st">"violetred"</span>), <span class="dt">bg=</span><span class="st">"gray92"</span>)</a></code></pre></div>
<p><img src="../../../Box%20Sync/Rpkgs/qtl2pleio/man/figures/README-1d-lod-plots-1.png" width="100%"></p>
<p>We see that the two traits share a peak on Chr 3.</p>
<p>And here are the observed QTL peaks with LOD &gt; 8.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/qtl2/topics/find_peaks">find_peaks</a></span>(s1, <span class="dt">map =</span> pm, <span class="dt">threshold=</span><span class="dv">8</span>)</a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="co">#&gt;   lodindex lodcolumn chr       pos       lod</span></a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="co">#&gt; 1        1       tr1   3 82.778059 24.552172</span></a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="co">#&gt; 2        2       tr2   2  3.164247  8.195879</span></a>
<a class="sourceLine" id="cb14-5" data-line-number="5"><span class="co">#&gt; 3        2       tr2   3 82.778059 17.454825</span></a>
<a class="sourceLine" id="cb14-6" data-line-number="6"><span class="co">#&gt; 4        2       tr2   X 95.069663 14.110670</span></a></code></pre></div>
</div>
<div id="perform-two-dimensional-scan-as-first-step-in-pleiotropy-vs-separate-qtl-hypothesis-test" class="section level3">
<h3 class="hasAnchor">
<a href="#perform-two-dimensional-scan-as-first-step-in-pleiotropy-vs-separate-qtl-hypothesis-test" class="anchor"></a>Perform two-dimensional scan as first step in pleiotropy vs. separate QTL hypothesis test</h3>
<p>We now have the inputs that we need to do a pleiotropy vs. separate QTL test. We have the founder allele dosages for one chromosome, <em>i.e.</em>, Chr 3, in the R object <code>pp</code>, the matrix of two trait measurements in <code>Y</code>, and a LOCO-derived kinship matrix, <code>kinship[[2]]</code>. We also specify, via the <code>start_snp</code> argument, the starting point for the two-dimensional scan within the array of founder allele dosages. Here, we choose the 38th marker in the array as the starting point. Via the <code>n_snp</code> argument, we specify the number of markers to include in the two-dimensional scan. Here, we input 25, so that we fit the bivariate linear mixed effects model at 25*25 = 625 ordered pairs of markers. In practice, we usually use between 100 and 300 markers for most two-dimensional scans.</p>
<p>Lastly, we specify the number of cores to use, with the <code>n_cores</code> argument. We set it to 1 here, to ensure that the vignette can be run by CRAN. However, in practice, you may wish to increase the number of cores to accelerate computing.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">out &lt;-<span class="st"> </span><span class="kw"><a href="reference/scan_pvl.html">scan_pvl</a></span>(<span class="dt">probs =</span> pp,</a>
<a class="sourceLine" id="cb15-2" data-line-number="2">                <span class="dt">pheno =</span> Y,</a>
<a class="sourceLine" id="cb15-3" data-line-number="3">                <span class="dt">kinship =</span> kinship[[<span class="dv">2</span>]], <span class="co"># 2nd entry in kinship list is Chr 3</span></a>
<a class="sourceLine" id="cb15-4" data-line-number="4">                <span class="dt">start_snp =</span> <span class="dv">38</span>,</a>
<a class="sourceLine" id="cb15-5" data-line-number="5">                <span class="dt">n_snp =</span> <span class="dv">25</span>, <span class="dt">n_cores =</span> <span class="dv">1</span></a>
<a class="sourceLine" id="cb15-6" data-line-number="6">                )</a></code></pre></div>
<p>The number of cores available will vary by computer. For example, on my Macbook pro computer, with 16GB RAM, I have access to 8 cores. If I use all 8, I can’t do other computing tasks, so I often set <code>n_cores</code> to 7.</p>
<p>To check how many cores are available on your computer, run this code.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">parallel<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/parallel/topics/detectCores">detectCores</a></span>()</a></code></pre></div>
<div id="create-a-profile-lod-plot-to-visualize-results-of-two-dimensional-scan" class="section level4">
<h4 class="hasAnchor">
<a href="#create-a-profile-lod-plot-to-visualize-results-of-two-dimensional-scan" class="anchor"></a>Create a profile LOD plot to visualize results of two-dimensional scan</h4>
<p>To visualize results from our two-dimensional scan, we calculate profile LOD for each trait. The code below makes use of the R package <code>ggplot2</code> to plot profile LODs over the scan region.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr)</a>
<a class="sourceLine" id="cb17-2" data-line-number="2">out <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="reference/tidy_scan_pvl.html">tidy_scan_pvl</a></span>(pm[[<span class="dv">2</span>]]) <span class="op">%&gt;%</span><span class="st"> </span><span class="co"># pm[[2]] is physical map for Chr 3</span></a>
<a class="sourceLine" id="cb17-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="reference/add_intercepts.html">add_intercepts</a></span>(<span class="dt">intercepts_univariate =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="fl">82.8</span>, <span class="fl">82.8</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb17-5" data-line-number="5"><span class="st">  </span><span class="kw"><a href="reference/plot_pvl.html">plot_pvl</a></span>(<span class="dt">phenames =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"tr1"</span>, <span class="st">"tr2"</span>))</a></code></pre></div>
<p><img src="../../../Box%20Sync/Rpkgs/qtl2pleio/man/figures/README-profile-plot-1.png" width="100%"></p>
</div>
<div id="calculate-the-likelihood-ratio-test-statistic-for-pleiotropy-v-separate-qtl" class="section level4">
<h4 class="hasAnchor">
<a href="#calculate-the-likelihood-ratio-test-statistic-for-pleiotropy-v-separate-qtl" class="anchor"></a>Calculate the likelihood ratio test statistic for pleiotropy v separate QTL</h4>
<p>We use the function <code>calc_lrt_tib</code> to calculate the likelihood ratio test statistic value for the specified traits and specified genomic region.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">(lrt &lt;-<span class="st"> </span><span class="kw"><a href="reference/calc_lrt_tib.html">calc_lrt_tib</a></span>(out))</a>
<a class="sourceLine" id="cb18-2" data-line-number="2"><span class="co">#&gt; [1] 0.7730965</span></a></code></pre></div>
</div>
</div>
<div id="bootstrap-analysis-to-get-p-values" class="section level3">
<h3 class="hasAnchor">
<a href="#bootstrap-analysis-to-get-p-values" class="anchor"></a>Bootstrap analysis to get p-values</h3>
<p>The calibration of test statistic values to get p-values uses bootstrap methods because we don’t know the theoretical distribution of the test statistic under the null hypothesis. Thus, we use a bootstrap approach to obtain an empirical distribution of test statistic values under the null hypothesis of the presence of one pleiotropic locus.</p>
<p>We will use the function <code>boot_pvl</code> from our package <code>qtl2pleio</code>.</p>
<p>We use a parametric bootstrap strategy in which we first use the studied phenotypes to infer the values of model parameters. Once we have the inferred values of the model parameters, we simulate phenotypes from the pleiotropy model (with the inferred parameter values).</p>
<p>A natural question that arises is “which marker’s allele probabilities do we use when simulating phenotypes?” We use the marker that, under the null hypothesis, <em>i.e.</em>, under the pleiotropy constraint, yields the greatest value of the log-likelihood.</p>
<p>Before we call <code>boot_pvl</code>, we need to identify the index (on the chromosome under study) of the marker that maximizes the likelihood under the pleiotropy constraint. To do this, we use the <code>qtl2pleio</code> function <code>find_pleio_peak_tib</code>.</p>
<<<<<<< HEAD
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">(pleio_index &lt;-<span class="st"> </span><span class="kw"><a href="reference/find_pleio_peak_tib.html">find_pleio_peak_tib</a></span>(out, <span class="dt">start_snp =</span> <span class="dv">38</span>))</a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="co">#&gt; loglik13 </span></a>
<a class="sourceLine" id="cb19-3" data-line-number="3"><span class="co">#&gt;       50</span></a></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">2018-11-25</span>)</a>
<a class="sourceLine" id="cb20-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.time">system.time</a></span>(b_out &lt;-<span class="st"> </span><span class="kw"><a href="reference/boot_pvl.html">boot_pvl</a></span>(<span class="dt">probs =</span> pp,</a>
<a class="sourceLine" id="cb20-3" data-line-number="3">         <span class="dt">pheno =</span> Y,</a>
<a class="sourceLine" id="cb20-4" data-line-number="4">         <span class="dt">pleio_peak_index =</span> pleio_index,</a>
<a class="sourceLine" id="cb20-5" data-line-number="5">         <span class="dt">kinship =</span> kinship[[<span class="dv">2</span>]], <span class="co"># 2nd element in kinship list is Chr 3</span></a>
<a class="sourceLine" id="cb20-6" data-line-number="6">         <span class="dt">nboot_per_job =</span> <span class="dv">10</span>,</a>
<a class="sourceLine" id="cb20-7" data-line-number="7">         <span class="dt">start_snp =</span> <span class="dv">38</span>,</a>
<a class="sourceLine" id="cb20-8" data-line-number="8">         <span class="dt">n_snp =</span> <span class="dv">25</span></a>
<a class="sourceLine" id="cb20-9" data-line-number="9">         ))</a>
<a class="sourceLine" id="cb20-10" data-line-number="10"><span class="co">#&gt;    user  system elapsed </span></a>
<a class="sourceLine" id="cb20-11" data-line-number="11"><span class="co">#&gt; 198.271   3.021 202.217</span></a></code></pre></div>
=======
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">(pleio_index &lt;-<span class="st"> </span><span class="kw"><a href="reference/find_pleio_peak_tib.html">find_pleio_peak_tib</a></span>(out, <span class="dt">start_snp =</span> <span class="dv">38</span>))</a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="co">#&gt; loglik13 </span></a>
<a class="sourceLine" id="cb16-3" data-line-number="3"><span class="co">#&gt;       50</span></a></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">2018-11-25</span>)</a>
<a class="sourceLine" id="cb17-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.time">system.time</a></span>(b_out &lt;-<span class="st"> </span><span class="kw"><a href="reference/boot_pvl.html">boot_pvl</a></span>(<span class="dt">probs =</span> pp,</a>
<a class="sourceLine" id="cb17-3" data-line-number="3">         <span class="dt">pheno =</span> Y, </a>
<a class="sourceLine" id="cb17-4" data-line-number="4">         <span class="dt">pleio_peak_index =</span> pleio_index, </a>
<a class="sourceLine" id="cb17-5" data-line-number="5">         <span class="dt">kinship =</span> kinship[[<span class="dv">2</span>]], <span class="co"># 2nd element in kinship list is Chr 3 </span></a>
<a class="sourceLine" id="cb17-6" data-line-number="6">         <span class="dt">nboot_per_job =</span> <span class="dv">10</span>, </a>
<a class="sourceLine" id="cb17-7" data-line-number="7">         <span class="dt">start_snp =</span> <span class="dv">38</span>, </a>
<a class="sourceLine" id="cb17-8" data-line-number="8">         <span class="dt">n_snp =</span> <span class="dv">25</span></a>
<a class="sourceLine" id="cb17-9" data-line-number="9">         ))</a>
<a class="sourceLine" id="cb17-10" data-line-number="10"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb17-11" data-line-number="11"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb17-12" data-line-number="12"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb17-13" data-line-number="13"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb17-14" data-line-number="14"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb17-15" data-line-number="15"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb17-16" data-line-number="16"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb17-17" data-line-number="17"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb17-18" data-line-number="18"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb17-19" data-line-number="19"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb17-20" data-line-number="20"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb17-21" data-line-number="21"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb17-22" data-line-number="22"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb17-23" data-line-number="23"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb17-24" data-line-number="24"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb17-25" data-line-number="25"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb17-26" data-line-number="26"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb17-27" data-line-number="27"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb17-28" data-line-number="28"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb17-29" data-line-number="29"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb17-30" data-line-number="30"><span class="co">#&gt;    user  system elapsed </span></a>
<a class="sourceLine" id="cb17-31" data-line-number="31"><span class="co">#&gt; 188.769   2.243 191.208</span></a></code></pre></div>
>>>>>>> a986bac8604684d1d721a3adcc4fdd77694710c4
<p>The argument <code>nboot_per_job</code> indicates the number of bootstrap samples that will be created and analyzed. Here, we set <code>nboot_per_job = 10</code>, so we expect to see returned a numeric vector of length 10, where each entry is a LRT statistic value from a distinct bootstrap sample.</p>
<p>Finally, we determine a bootstrap p-value in the usual method. We treat the bootstrap samples’ test statistics as an empirical distribution of the test statistic under the null hypothesis of pleiotropy. Thus, to get a p-value, we want to ask “What is the probability, under the null hypothesis, of observing a test statistic value that is at least as extreme as that which we observed?”</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">b_out</a>
<a class="sourceLine" id="cb21-2" data-line-number="2"><span class="co">#&gt;  [1] 0.0000000 0.0000000 0.4512753 0.0000000 0.0000000 0.0000000 0.0000000</span></a>
<a class="sourceLine" id="cb21-3" data-line-number="3"><span class="co">#&gt;  [8] 0.0000000 0.0000000 0.9867788</span></a>
<a class="sourceLine" id="cb21-4" data-line-number="4">(pvalue &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/mean">mean</a></span>(b_out <span class="op">&gt;=</span><span class="st"> </span>lrt))</a>
<a class="sourceLine" id="cb21-5" data-line-number="5"><span class="co">#&gt; [1] 0.1</span></a></code></pre></div>
<p>In practice, one would want to use many more bootstrap samples to achieve an empirical distribution that is closer to the theoretical distribution of the test statistic under the null hypothesis.</p>
<p>However, if one wants to perform analyses with a reasonable number - say 400 - bootstrap samples, this will take a very long time - many days - on a single laptop computer. We have used a series of computer clusters that are coordinated by the University of Wisconsin-Madison’s Center for High-throughput Computing (<a href="http://chtc.cs.wisc.edu" class="uri">http://chtc.cs.wisc.edu</a>). We typically are able to analyze 1000 bootstrap samples in less than 24 hours with this service.</p>
</div>
</div>
<div id="code-of-conduct" class="section level2">
<h2 class="hasAnchor">
<a href="#code-of-conduct" class="anchor"></a>Code of Conduct</h2>
<p>Please note that this project is released with a <a href="CONDUCT.html">Contributor Code of Conduct</a>. By participating in this project you agree to abide by its terms.</p>
</div>
<div id="citation-information" class="section level2">
<h2 class="hasAnchor">
<a href="#citation-information" class="anchor"></a>Citation information</h2>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/citation">citation</a></span>(<span class="st">"qtl2pleio"</span>)</a>
<a class="sourceLine" id="cb22-2" data-line-number="2"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb22-3" data-line-number="3"><span class="co">#&gt; To cite qtl2pleio in publications use:</span></a>
<a class="sourceLine" id="cb22-4" data-line-number="4"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb22-5" data-line-number="5"><span class="co">#&gt;   Frederick Boehm (2019). qtl2pleio: Testing pleiotropy vs.</span></a>
<a class="sourceLine" id="cb22-6" data-line-number="6"><span class="co">#&gt;   separate QTL in multiparental populations. R package version</span></a>
<a class="sourceLine" id="cb22-7" data-line-number="7"><span class="co">#&gt;   0.1.2.9000. URL https://github.com/fboehm/qtl2pleio</span></a>
<a class="sourceLine" id="cb22-8" data-line-number="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb22-9" data-line-number="9"><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></a>
<a class="sourceLine" id="cb22-10" data-line-number="10"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb22-11" data-line-number="11"><span class="co">#&gt;   @Manual{,</span></a>
<a class="sourceLine" id="cb22-12" data-line-number="12"><span class="co">#&gt;     title = {qtl2pleio: Testing pleiotropy vs. separate QTL in multiparental populations},</span></a>
<a class="sourceLine" id="cb22-13" data-line-number="13"><span class="co">#&gt;     author = {Frederick Boehm},</span></a>
<a class="sourceLine" id="cb22-14" data-line-number="14"><span class="co">#&gt;     year = {2019},</span></a>
<a class="sourceLine" id="cb22-15" data-line-number="15"><span class="co">#&gt;     note = {R package version 0.1.2.9000},</span></a>
<a class="sourceLine" id="cb22-16" data-line-number="16"><span class="co">#&gt;     url = {https://github.com/fboehm/qtl2pleio},</span></a>
<a class="sourceLine" id="cb22-17" data-line-number="17"><span class="co">#&gt;   }</span></a></code></pre></div>
<p>Icon image of mice: <a href="https://www.jax.org/~/media/JaxWeb/images/jax-mice-and-services/mice/datasheets/002468" class="uri">https://www.jax.org/~/media/JaxWeb/images/jax-mice-and-services/mice/datasheets/002468</a></p>
<p>Icon image of Arabidopsis: <a href="https://sites.cns.utexas.edu/sites/default/files/styles/os_files_large/public/juenger_lab/files/pp1244cvr001-1.gif" class="uri">https://sites.cns.utexas.edu/sites/default/files/styles/os_files_large/public/juenger_lab/files/pp1244cvr001-1.gif</a></p>
<p>Icon image of Drosophila: <a href="https://www.yourgenome.org/sites/default/files/styles/banner/public/banners/stories/fruit-flies-in-the-laboratory/single-fruit-fly-drosophila-melanogaster-on-white-background-cropped.jpg" class="uri">https://www.yourgenome.org/sites/default/files/styles/banner/public/banners/stories/fruit-flies-in-the-laboratory/single-fruit-fly-drosophila-melanogaster-on-white-background-cropped.jpg</a></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/fboehm/qtl2pleio">https://​github.com/​fboehm/​qtl2pleio</a>
</li>
<li>Report a bug at <br><a href="https://github.com/fboehm/qtl2pleio/issues">https://​github.com/​fboehm/​qtl2pleio/​issues</a>
</li>
<li>Frederick Boehm's Website at <br><a href="https://fboehm.us">https://​fboehm.us</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing qtl2pleio</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Frederick Boehm <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-1644-5931" target="orcid.widget"><img src="https://members.orcid.org/sites/default/files/vector_iD_icon.svg" class="orcid" alt="ORCID" height="16"></a> </li>
</ul>
</div>

      <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="http://mybinder.org/v2/gh/fboehm/qtl2pleio/master?urlpath=rstudio"><img src="http://mybinder.org/badge_logo.svg" alt="Binder"></a></li>
<li><a href="https://travis-ci.org/fboehm/qtl2pleio"><img src="https://travis-ci.org/fboehm/qtl2pleio.svg?branch=master" alt="Travis-CI Build Status"></a></li>
<li><a href="https://codecov.io/github/fboehm/qtl2pleio?branch=master"><img src="https://img.shields.io/codecov/c/github/fboehm/qtl2pleio/master.svg" alt="Coverage Status"></a></li>
<li><a href="https://www.repostatus.org/#wip"><img src="https://www.repostatus.org/badges/latest/wip.svg" alt="Project Status: WIP – Initial development is in progress, but there has not yet been a stable, usable release suitable for the public."></a></li>
</ul>
</div>
</div>

</div>


      <footer><div class="copyright">
  <p>Developed by Frederick Boehm.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9000.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
