<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Testing pleiotropy vs. separate QTL in multiparental populations • qtl2pleio</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Testing pleiotropy vs. separate QTL in multiparental populations">
<meta property="og:description" content="The package contains code that implements an
    adaptation of Jiang &amp; Zeng's (1995) likelihood ratio test for testing
    the null hypothesis of pleiotropy against the alternative hypothesis,
    two separate QTL. The test differs from that in Jiang &amp; Zeng (1995) in
    that our test accommodates multiparental populations, such as the Diversity 
    Outbred mouse population.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">qtl2pleio</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/bootstrap-cluster.html">Using HTCondor for bootstrap analysis</a>
    </li>
    <li>
      <a href="articles/recla-analysis.html">Recla Analysis</a>
    </li>
    <li>
      <a href="articles/testing-pleiotropy-vs-separate-qtl.html">Testing pleiotropy v separate QTL in multiparental populations</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/fboehm/qtl2pleio">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    

    
    
<!-- README.md is generated from README.Rmd. Please edit README.Rmd -->
<div id="qtl2pleio" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#qtl2pleio" class="anchor"></a>qtl2pleio</h1></div>

<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p><code>qtl2pleio</code> is a software package for use with the <a href="https://cran.r-project.org/">R statistical computing environment</a>. <code>qtl2pleio</code> is freely available for download and use. I share it under the <a href="https://opensource.org/licenses/mit-license.php">MIT license</a>. The user will also want to download and install the <a href="https://kbroman.org/qtl2/"><code>qtl2</code> R package</a>.</p>
<p>Click <a href="https://mybinder.org/v2/gh/fboehm/qtl2pleio/master?urlpath=rstudio">here</a> to explore <code>qtl2pleio</code> within a live <a href="https://www.rstudio.com/">Rstudio</a> session in “the cloud”.</p>
</div>
<div id="goals" class="section level2">
<h2 class="hasAnchor">
<a href="#goals" class="anchor"></a>Goals</h2>
<p>The goal of <code>qtl2pleio</code> is, for a pair of traits that show evidence for a QTL in a common region, to distinguish between pleiotropy (the null hypothesis, that they are affected by a common QTL) and the alternative that they are affected by separate QTL. It extends the likelihood ratio test of <a href="http://www.genetics.org/content/140/3/1111.long">Jiang and Zeng (1995)</a> for multiparental populations, such as Diversity Outbred mice, including the use of multivariate polygenic random effects to account for population structure. <code>qtl2pleio</code> data structures are those used in the <a href="https://kbroman.org/qtl2"><code>rqtl/qtl2</code></a> package.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To install qtl2pleio, use <code>install_github()</code> from the <a href="https://devtools.r-lib.org">devtools</a> package.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/install.packages">install.packages</a></span>(<span class="st">"devtools"</span>)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"fboehm/qtl2pleio"</span>)</a></code></pre></div>
<p>You may also wish to install <a href="https://kbroman.org/qtl2">R/qtl2</a> and the <a href="https://github.com/rqtl/qtl2convert"><code>qtl2convert</code></a> package. We will use both below.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/install.packages">install.packages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"qtl2"</span>, <span class="st">"qtl2convert"</span>), <span class="dt">repos=</span><span class="st">"http://rqtl.org/qtl2cran"</span>)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co">#&gt; The downloaded binary packages are in</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="co">#&gt;  /var/folders/wd/lxmyvz590xb81c5z1j88b3800000gn/T//RtmpBTfSre/downloaded_packages</span></a></code></pre></div>
</div>
<div id="example" class="section level2">
<h2 class="hasAnchor">
<a href="#example" class="anchor"></a>Example</h2>
<p>Below, we walk through an example analysis with Diversity Outbred mouse data. We need a number of preliminary steps before we can perform our test of pleiotropy vs. separate QTL. Many procedures rely on the R package <code>qtl2</code>. We first load the <code>qtl2</code>, <code>qtl2convert</code>, and <code>qtl2pleio</code> packages.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(qtl2)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(qtl2convert)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(qtl2pleio)</a></code></pre></div>
<div id="reading-data-from-qtl2data-repository-on-github" class="section level3">
<h3 class="hasAnchor">
<a href="#reading-data-from-qtl2data-repository-on-github" class="anchor"></a>Reading data from <code>qtl2data</code> repository on github</h3>
<p>We’ll consider the <a href="https://github.com/rqtl/qtl2data/tree/master/DOex"><code>DOex</code></a> data in the <a href="https://github.com/rqtl/qtl2data"><code>qtl2data</code></a> repository. We’ll download the DOex.zip file before calculating founder allele dosages.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">file &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(<span class="st">"https://raw.githubusercontent.com/rqtl/"</span>,</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">               <span class="st">"qtl2data/master/DOex/DOex.zip"</span>)</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">DOex &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/qtl2/topics/read_cross2">read_cross2</a></span>(file)</a></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">probs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/qtl2/topics/calc_genoprob">calc_genoprob</a></span>(DOex)</a></code></pre></div>
<p>Let’s calculate the founder allele dosages from the 36-state genotype probabilities.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">pr &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/qtl2/topics/genoprob_to_alleleprob">genoprob_to_alleleprob</a></span>(probs)</a></code></pre></div>
<p>We now have an allele probabilities object stored in <code>pr</code>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(pr)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="co">#&gt; [1] "2" "3" "X"</span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/dim">dim</a></span>(pr<span class="op">$</span><span class="st">`</span><span class="dt">2</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="co">#&gt; [1] 261   8 127</span></a></code></pre></div>
<p>We see that <code>pr</code> is a list of 3 three-dimensional arrays - one array for each of 3 chromosomes.</p>
</div>
<div id="kinship-calculations" class="section level3">
<h3 class="hasAnchor">
<a href="#kinship-calculations" class="anchor"></a>Kinship calculations</h3>
<p>For our statistical model, we need a kinship matrix. We get one with the <code>calc_kinship</code> function in the <code>rqtl/qtl2</code> package.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">kinship &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/qtl2/topics/calc_kinship">calc_kinship</a></span>(<span class="dt">probs =</span> pr, <span class="dt">type =</span> <span class="st">"loco"</span>)</a></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/str">str</a></span>(kinship)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="co">#&gt; List of 3</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="co">#&gt;  $ 2: num [1:261, 1:261] 0.6934 0.0705 0.2356 0.0558 0.0513 ...</span></a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="co">#&gt;   ..- attr(*, "n_pos")= int 195</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span></a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="co">#&gt;   .. ..$ : chr [1:261] "1" "4" "5" "6" ...</span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="co">#&gt;   .. ..$ : chr [1:261] "1" "4" "5" "6" ...</span></a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="co">#&gt;  $ 3: num [1:261, 1:261] 0.6662 0.0647 0.2024 0.1129 0.0772 ...</span></a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="co">#&gt;   ..- attr(*, "n_pos")= int 220</span></a>
<a class="sourceLine" id="cb9-10" data-line-number="10"><span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span></a>
<a class="sourceLine" id="cb9-11" data-line-number="11"><span class="co">#&gt;   .. ..$ : chr [1:261] "1" "4" "5" "6" ...</span></a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="co">#&gt;   .. ..$ : chr [1:261] "1" "4" "5" "6" ...</span></a>
<a class="sourceLine" id="cb9-13" data-line-number="13"><span class="co">#&gt;  $ X: num [1:261, 1:261] 0.4871 0.0831 0.1953 0.1043 0.1125 ...</span></a>
<a class="sourceLine" id="cb9-14" data-line-number="14"><span class="co">#&gt;   ..- attr(*, "n_pos")= int 229</span></a>
<a class="sourceLine" id="cb9-15" data-line-number="15"><span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span></a>
<a class="sourceLine" id="cb9-16" data-line-number="16"><span class="co">#&gt;   .. ..$ : chr [1:261] "1" "4" "5" "6" ...</span></a>
<a class="sourceLine" id="cb9-17" data-line-number="17"><span class="co">#&gt;   .. ..$ : chr [1:261] "1" "4" "5" "6" ...</span></a></code></pre></div>
</div>
<div id="statistical-model-specification" class="section level3">
<h3 class="hasAnchor">
<a href="#statistical-model-specification" class="anchor"></a>Statistical model specification</h3>
<p>We use the multivariate linear mixed effects model:</p>
<p><span class="math display">\[vec(Y) = X vec(B) + vec(G) + vec(E)\]</span></p>
<p>where <span class="math inline">\(Y\)</span> contains phenotypes, X contains founder allele probabilities and covariates, and B contains founder allele effects. G is the polygenic random effects, while E is the random errors. We provide more details in the vignette.</p>
</div>
<div id="simulating-phenotypes-with-qtl2pleiosim1" class="section level3">
<h3 class="hasAnchor">
<a href="#simulating-phenotypes-with-qtl2pleiosim1" class="anchor"></a>Simulating phenotypes with <code><a href="reference/sim1.html">qtl2pleio::sim1</a></code>
</h3>
<p>The function to simulate phenotypes in <code>qtl2pleio</code> is <code>sim1</code>.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="co"># set up the design matrix, X</span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2">pp &lt;-<span class="st"> </span>pr[[<span class="dv">2</span>]] <span class="co">#we'll work with Chr 3's genotype data</span></a></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="co">#Next, we prepare a design matrix X</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2">X &lt;-<span class="st"> </span>gemma2<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/gemma2/topics/stagger_mats">stagger_mats</a></span>(pp[ , , <span class="dv">50</span>], pp[ , , <span class="dv">50</span>])</a></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="co"># assemble B matrix of allele effects</span></a>
<a class="sourceLine" id="cb12-2" data-line-number="2">B &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">matrix</a></span>(<span class="dt">data =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="op">-</span><span class="dv">1</span>, <span class="dv">-1</span>, <span class="dv">-1</span>, <span class="dv">-1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">-1</span>, <span class="dv">-1</span>, <span class="dv">-1</span>, <span class="dv">-1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>), <span class="dt">nrow =</span> <span class="dv">8</span>, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">byrow =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="co"># set.seed to ensure reproducibility</span></a>
<a class="sourceLine" id="cb12-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">2018-01-30</span>)</a>
<a class="sourceLine" id="cb12-5" data-line-number="5"><span class="co"># call to sim1</span></a>
<a class="sourceLine" id="cb12-6" data-line-number="6">Ypre &lt;-<span class="st"> </span><span class="kw"><a href="reference/sim1.html">sim1</a></span>(<span class="dt">X =</span> X, <span class="dt">B =</span> B, <span class="dt">Vg =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/diag">diag</a></span>(<span class="dv">2</span>), <span class="dt">Ve =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/diag">diag</a></span>(<span class="dv">2</span>), <span class="dt">kinship =</span> kinship[[<span class="dv">2</span>]])</a>
<a class="sourceLine" id="cb12-7" data-line-number="7">Y &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">matrix</a></span>(Ypre, <span class="dt">nrow =</span> <span class="dv">261</span>, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">byrow =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb12-8" data-line-number="8"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(Y) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(pp)</a>
<a class="sourceLine" id="cb12-9" data-line-number="9"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(Y) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"tr1"</span>, <span class="st">"tr2"</span>)</a></code></pre></div>
<p>Let’s perform univariate QTL mapping for each of the two traits in the Y matrix.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">s1 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/qtl2/topics/scan1">scan1</a></span>(<span class="dt">genoprobs =</span> pr, <span class="dt">pheno =</span> Y, <span class="dt">kinship =</span> kinship)</a></code></pre></div>
<p>Here is a plot of the results.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(s1, DOex<span class="op">$</span>pmap)</a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(s1, DOex<span class="op">$</span>pmap, <span class="dt">lod=</span><span class="dv">2</span>, <span class="dt">col=</span><span class="st">"violetred"</span>, <span class="dt">add=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/legend">legend</a></span>(<span class="st">"topleft"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(s1), <span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">col=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"darkslateblue"</span>, <span class="st">"violetred"</span>), <span class="dt">bg=</span><span class="st">"gray92"</span>)</a></code></pre></div>
<p><img src="../../../Box%20Sync/Rpkgs/qtl2pleio/man/figures/README-1d-lod-plots-1.png" width="100%"></p>
<p>And here are the observed QTL peaks with LOD &gt; 8.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/qtl2/topics/find_peaks">find_peaks</a></span>(s1, <span class="dt">map =</span> DOex<span class="op">$</span>pmap, <span class="dt">threshold=</span><span class="dv">8</span>)</a>
<a class="sourceLine" id="cb15-2" data-line-number="2"><span class="co">#&gt;   lodindex lodcolumn chr       pos      lod</span></a>
<a class="sourceLine" id="cb15-3" data-line-number="3"><span class="co">#&gt; 1        1       tr1   3  82.77806 16.19704</span></a>
<a class="sourceLine" id="cb15-4" data-line-number="4"><span class="co">#&gt; 2        2       tr2   3  82.77806 18.26406</span></a>
<a class="sourceLine" id="cb15-5" data-line-number="5"><span class="co">#&gt; 3        2       tr2   X 103.79061 16.19708</span></a></code></pre></div>
</div>
<div id="perform-two-dimensional-scan-as-first-step-in-pleiotropy-vs-separate-qtl-hypothesis-test" class="section level3">
<h3 class="hasAnchor">
<a href="#perform-two-dimensional-scan-as-first-step-in-pleiotropy-vs-separate-qtl-hypothesis-test" class="anchor"></a>Perform two-dimensional scan as first step in pleiotropy vs. separate QTL hypothesis test</h3>
<p>We now have the inputs that we need to do a pleiotropy vs. separate QTL test. We have the founder allele dosages for one chromosome, <em>i.e.</em>, Chr 3, in the R object <code>pp</code>, the matrix of two trait measurements in <code>Y</code>, and a LOCO-derived kinship matrix, <code>kinship[[2]]</code>.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">out &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="reference/scan_pvl.html">scan_pvl</a></span>(<span class="dt">probs =</span> pp,</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">                <span class="dt">pheno =</span> Y,</a>
<a class="sourceLine" id="cb16-3" data-line-number="3">                <span class="dt">kinship =</span> kinship[[<span class="dv">2</span>]], <span class="co"># 2nd entry in kinship list is Chr 3</span></a>
<a class="sourceLine" id="cb16-4" data-line-number="4">                <span class="dt">start_snp =</span> <span class="dv">38</span>,</a>
<a class="sourceLine" id="cb16-5" data-line-number="5">                <span class="dt">n_snp =</span> <span class="dv">25</span>, <span class="dt">n_cores =</span> <span class="dv">1</span></a>
<a class="sourceLine" id="cb16-6" data-line-number="6">                ))</a></code></pre></div>
</div>
<div id="create-a-profile-lod-plot-to-visualize-results-of-two-dimensional-scan" class="section level3">
<h3 class="hasAnchor">
<a href="#create-a-profile-lod-plot-to-visualize-results-of-two-dimensional-scan" class="anchor"></a>Create a profile LOD plot to visualize results of two-dimensional scan</h3>
<p>To visualize results from our two-dimensional scan, we calculate profile LOD for each trait. The code below makes use of the R package <code>ggplot2</code> to plot profile LODs over the scan region.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr)</a>
<a class="sourceLine" id="cb17-2" data-line-number="2">out <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="reference/tidy_scan_pvl.html">tidy_scan_pvl</a></span>(DOex<span class="op">$</span>pmap[[<span class="dv">2</span>]]) <span class="op">%&gt;%</span><span class="st"> </span><span class="co"># pmap[[2]] is physical map for Chr 3</span></a>
<a class="sourceLine" id="cb17-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="reference/plot_pvl.html">plot_pvl</a></span>()</a></code></pre></div>
<p><img src="../../../Box%20Sync/Rpkgs/qtl2pleio/man/figures/README-profile-plot-1.png" width="100%"></p>
</div>
<div id="calculate-the-likelihood-ratio-test-statistic-for-pleiotropy-v-separate-qtl" class="section level3">
<h3 class="hasAnchor">
<a href="#calculate-the-likelihood-ratio-test-statistic-for-pleiotropy-v-separate-qtl" class="anchor"></a>Calculate the likelihood ratio test statistic for pleiotropy v separate QTL</h3>
<p>We use the function <code>calc_lrt_tib</code> to calculate the likelihood ratio test statistic value for the specified traits and specified genomic region.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">(lrt &lt;-<span class="st"> </span><span class="kw"><a href="reference/calc_lrt_tib.html">calc_lrt_tib</a></span>(out))</a>
<a class="sourceLine" id="cb18-2" data-line-number="2"><span class="co">#&gt; [1] 0</span></a></code></pre></div>
</div>
<div id="bootstrap-analysis-to-get-p-values" class="section level3">
<h3 class="hasAnchor">
<a href="#bootstrap-analysis-to-get-p-values" class="anchor"></a>Bootstrap analysis to get p-values</h3>
<p>Before we call <code>boot_pvl</code>, we need to identify the index (on the chromosome under study) of the marker that maximizes the likelihood under the pleiotropy constraint. To do this, we use the <code>qtl2pleio</code> function <code>find_pleio_peak_tib</code>.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">(pleio_index &lt;-<span class="st"> </span><span class="kw"><a href="reference/find_pleio_peak_tib.html">find_pleio_peak_tib</a></span>(out, <span class="dt">start_snp =</span> <span class="dv">38</span>))</a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="co">#&gt; loglik13 </span></a>
<a class="sourceLine" id="cb19-3" data-line-number="3"><span class="co">#&gt;       50</span></a></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">2018-11-25</span>) <span class="co"># set for reproducibility purposes.</span></a>
<a class="sourceLine" id="cb20-2" data-line-number="2">b_out &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/message">suppressMessages</a></span>(<span class="kw"><a href="reference/boot_pvl.html">boot_pvl</a></span>(<span class="dt">probs =</span> pp,</a>
<a class="sourceLine" id="cb20-3" data-line-number="3">         <span class="dt">pheno =</span> Y,</a>
<a class="sourceLine" id="cb20-4" data-line-number="4">         <span class="dt">pleio_peak_index =</span> pleio_index,</a>
<a class="sourceLine" id="cb20-5" data-line-number="5">         <span class="dt">kinship =</span> kinship[[<span class="dv">2</span>]], <span class="co"># 2nd element in kinship list is Chr 3</span></a>
<a class="sourceLine" id="cb20-6" data-line-number="6">         <span class="dt">nboot_per_job =</span> <span class="dv">10</span>,</a>
<a class="sourceLine" id="cb20-7" data-line-number="7">         <span class="dt">start_snp =</span> <span class="dv">38</span>,</a>
<a class="sourceLine" id="cb20-8" data-line-number="8">         <span class="dt">n_snp =</span> <span class="dv">25</span></a>
<a class="sourceLine" id="cb20-9" data-line-number="9">         ))</a></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">(pvalue &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/mean">mean</a></span>(b_out <span class="op">&gt;=</span><span class="st"> </span>lrt))</a>
<a class="sourceLine" id="cb21-2" data-line-number="2"><span class="co">#&gt; [1] 1</span></a></code></pre></div>
</div>
</div>
<div id="code-of-conduct" class="section level2">
<h2 class="hasAnchor">
<a href="#code-of-conduct" class="anchor"></a>Code of Conduct</h2>
<p>Please note that this project is released with a <a href="CONDUCT.html">Contributor Code of Conduct</a>. By participating in this project you agree to abide by its terms.</p>
</div>
<div id="citation-information" class="section level2">
<h2 class="hasAnchor">
<a href="#citation-information" class="anchor"></a>Citation information</h2>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/citation">citation</a></span>(<span class="st">"qtl2pleio"</span>)</a>
<a class="sourceLine" id="cb22-2" data-line-number="2"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb22-3" data-line-number="3"><span class="co">#&gt; To cite qtl2pleio in publications use:</span></a>
<a class="sourceLine" id="cb22-4" data-line-number="4"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb22-5" data-line-number="5"><span class="co">#&gt;   Boehm FJ, Chesler EJ, Yandell BS, Broman KW (2019) Testing</span></a>
<a class="sourceLine" id="cb22-6" data-line-number="6"><span class="co">#&gt;   pleiotropy vs. separate QTL in multiparental populations Biorxiv</span></a>
<a class="sourceLine" id="cb22-7" data-line-number="7"><span class="co">#&gt;   https://www.biorxiv.org/content/10.1101/550939v1</span></a>
<a class="sourceLine" id="cb22-8" data-line-number="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb22-9" data-line-number="9"><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></a>
<a class="sourceLine" id="cb22-10" data-line-number="10"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb22-11" data-line-number="11"><span class="co">#&gt;   @Article{Boehm2019testing,</span></a>
<a class="sourceLine" id="cb22-12" data-line-number="12"><span class="co">#&gt;     title = {Testing pleiotropy vs. separate QTL in multiparental populations},</span></a>
<a class="sourceLine" id="cb22-13" data-line-number="13"><span class="co">#&gt;     author = {Frederick J. Boehm and Elissa J. Chesler and Brian S. Yandell and Karl W. Broman},</span></a>
<a class="sourceLine" id="cb22-14" data-line-number="14"><span class="co">#&gt;     journal = {Biorxiv},</span></a>
<a class="sourceLine" id="cb22-15" data-line-number="15"><span class="co">#&gt;     year = {2019},</span></a>
<a class="sourceLine" id="cb22-16" data-line-number="16"><span class="co">#&gt;     url = {https://www.biorxiv.org/content/10.1101/550939v1},</span></a>
<a class="sourceLine" id="cb22-17" data-line-number="17"><span class="co">#&gt;     eprint = {https://www.biorxiv.org/content/early/2019/02/15/550939.full.pdf},</span></a>
<a class="sourceLine" id="cb22-18" data-line-number="18"><span class="co">#&gt;   }</span></a></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/fboehm/qtl2pleio">https://​github.com/​fboehm/​qtl2pleio</a>
</li>
<li>Report a bug at <br><a href="https://github.com/fboehm/qtl2pleio/issues">https://​github.com/​fboehm/​qtl2pleio/​issues</a>
</li>
<li>Frederick Boehm's Website at <br><a href="https://fboehm.us">https://​fboehm.us</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing qtl2pleio</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Frederick Boehm <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-1644-5931" target="orcid.widget"><img src="https://members.orcid.org/sites/default/files/vector_iD_icon.svg" class="orcid" alt="ORCID" height="16"></a> </li>
</ul>
</div>

      <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://mybinder.org/v2/gh/fboehm/qtl2pleio/master?urlpath=rstudio"><img src="https://mybinder.org/badge_logo.svg" alt="Binder"></a></li>
<li><a href="https://travis-ci.org/fboehm/qtl2pleio"><img src="https://travis-ci.org/fboehm/qtl2pleio.svg?branch=master" alt="Travis-CI Build Status"></a></li>
<li><a href="https://codecov.io/github/fboehm/qtl2pleio?branch=master"><img src="https://img.shields.io/codecov/c/github/fboehm/qtl2pleio/master.svg" alt="Coverage Status"></a></li>
<li><a href="https://www.repostatus.org/#wip"><img src="https://www.repostatus.org/badges/latest/wip.svg" alt="Project Status: WIP – Initial development is in progress, but there has not yet been a stable, usable release suitable for the public."></a></li>
</ul>
</div>
</div>

</div>


      <footer><div class="copyright">
  <p>Developed by Frederick Boehm.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9000.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
