<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using HTCondor for bootstrap analysis • qtl2pleio</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Using HTCondor for bootstrap analysis">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">qtl2pleio</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/bootstrap-cluster.html">Using HTCondor for bootstrap analysis</a>
    </li>
    <li>
      <a href="../articles/recla-analysis.html">Recla Analysis</a>
    </li>
    <li>
      <a href="../articles/testing-pleiotropy-vs-separate-qtl.html">Testing pleiotropy v separate QTL in multiparental populations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/fboehm/qtl2pleio">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Using HTCondor for bootstrap analysis</h1>
                        <h4 class="author">Frederick Boehm</h4>
            
            <h4 class="date">2019-04-28</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/fboehm/qtl2pleio/blob/master/vignettes/bootstrap-cluster.Rmd"><code>vignettes/bootstrap-cluster.Rmd</code></a></small>
      <div class="hidden name"><code>bootstrap-cluster.Rmd</code></div>

    </div>

    
    
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>Performing a bootstrap analysis with <code>qtl2pleio</code> requires first sampling a (bivariate) phenotype from the inferred bivariate normal distribution, then a two-dimensional QTL scan over the defined scan region. Often, one wants to include more than 100 markers in a scan region. Each two-dimensional scan can take tens of minutes with a single core. Thus, performing a bootstrap analysis with, say, 1000 bootstrap samples is prohibitively time-consuming on most computers. For this reason, we used a high-throughput computing cluster at the University of Wisconsin-Madison (where <code>qtl2pleio</code> was developed). The UW-Madison Center for High-Throughput Computing (CHTC) is available to all UW-Madison researchers. While we recognize that <code>qtl2pleio</code> users may not have access to the CHTC, we anticipate that some will have access to computing clusters. Hopefully this vignette can be adapted to your needs.</p>
</div>
<div id="directory-structure-files-setup-for-using-condor" class="section level2">
<h2 class="hasAnchor">
<a href="#directory-structure-files-setup-for-using-condor" class="anchor"></a>Directory structure, files &amp; setup for using Condor</h2>
<p>We placed on github a repository that contains the results from our bootstrap analysis of the Recla data.</p>
<p>Here is the repository’s url: <a href="https://github.com/fboehm/qtl2pleio-manuscript-chtc/" class="uri">https://github.com/fboehm/qtl2pleio-manuscript-chtc/</a></p>
<p>Within this repo are three subdirectories. We’ll examine in detail the “Recla-bootstrap” subdirectory.</p>
<p>Within this directory, we created four subdirectories:</p>
<ol style="list-style-type: decimal">
<li>Rscript</li>
<li>data</li>
<li>shell_scripts</li>
<li>submit_files</li>
</ol>
<p>The Github repository also contains a fifth subdirectory, <code>squid</code> and a sixth subdirectory <code>results</code>, that are included for completeness. In practice, files in <code>squid</code> were in a distinct directory. We describe it below.</p>
<div id="files-in-rscript" class="section level3">
<h3 class="hasAnchor">
<a href="#files-in-rscript" class="anchor"></a>Files in <code>Rscript</code>
</h3>
<p>The <code>Rscript</code> subdirectory contains a file, <code>boot-Recla-10-22.R</code> that contains R code for our bootstrap analysis.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">fn &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/file.path">file.path</a></span>(<span class="st">"https://raw.githubusercontent.com"</span>, <span class="st">"fboehm/qtl2pleio-manuscript/master/chtc"</span>, </a>
<a class="sourceLine" id="cb1-2" data-line-number="2">    <span class="st">"Recla-bootstrap/Rscript/boot-Recla-10-22.R"</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">foo &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/readLines">readLines</a></span>(fn)</a></code></pre></div>
<p>Here we display the contents of our <code>boot-Recla-10-22.R</code> file.</p>
<pre><code>##
pleio_peak_index &lt;- 788
##



library(qtl2pleio)
library(stringr)
library(dplyr)
library(readr)
args &lt;- R.utils::commandArgs(trailingOnly = TRUE, asValues = TRUE)
# translate command line args
print(args)
print(args$argname)
(proc_num &lt;- as.numeric(args$argname))
(run_num &lt;- as.numeric(args$run_num))
(nboot_per_job &lt;- args$nboot_per_job) # think 1 boot per job
(nsnp &lt;- as.integer(args$nsnp))
(s1 &lt;- as.integer(args$s1))
###############
library(qtl2)
recla &lt;- readRDS("data/recla.rds")
# make sex a covariate for use in pvl_scan
recla[[6]][ , 1, drop = FALSE] -&gt; sex
sex &lt;- sex == "female"
# insert pseudomarkers
insert_pseudomarkers(recla, step = 0.1) -&gt; pseudomap

## ------------------------------------------------------------------------
## ------------------------------------------------------------------------
pp &lt;- readRDS("data/recla-aprobs-chr8.rds")

## ------------------------------------------------------------------------
kinship &lt;- readRDS("data/recla-kinship.rds")

## ------------------------------------------------------------------------
recla$pheno -&gt; ph
log(ph) -&gt; lph
apply(FUN = broman::winsorize, X = lph, MARGIN = 2) -&gt; wlph

## ------------------------------------------------------------------------
phe &lt;- wlph[, c(10, 22)]
gm &lt;- pseudomap$`8`
k &lt;- kinship[[8]]
## ------------------------------------------------------------------------
library(qtl2pleio)


###############
# simulate a phenotype
X1 &lt;- pp[ , , pleio_peak_index]
cbind(X1, unlist(sex)) -&gt; Xpre
## remove subjects with missing values of phenotype
is.na(phe[ , 1]) | is.na(phe[ , 2]) -&gt; missing_indic
phe_nona &lt;- phe[!missing_indic, ]
Xpre_nona &lt;- Xpre[!missing_indic, ]
k_nona &lt;- k[!missing_indic, !missing_indic]
##
gemma2::stagger_mats(Xpre_nona, Xpre_nona) -&gt; X
set.seed(proc_num)
calc_covs(pheno = phe_nona, kinship = k_nona) -&gt; cc_out
(cc_out$Vg -&gt; Vg)
(cc_out$Ve -&gt; Ve)
# calculate Sigma
calc_Sigma(Vg = Vg, Ve = Ve, K = k_nona) -&gt; Sigma
solve(Sigma) -&gt; Sigma_inv
# calc Bhat 
B &lt;- calc_Bhat(X = X, Sigma_inv = Sigma_inv, Y = phe_nona)
# Start loop
lrt &lt;- numeric()
for (i in 1:nboot_per_job){
  sim1(X = X, B = B, Vg = Vg, Ve = Ve, kinship = k_nona) -&gt; foo
  matrix(foo, ncol = 2, byrow = FALSE) -&gt; Ysim
  rownames(Ysim) &lt;- rownames(phe_nona)
  colnames(Ysim) &lt;- c("t1", "t2")
  scan_pvl(probs = pp[!missing_indic, , ], pheno = Ysim, covariates = sex[!missing_indic, , drop = FALSE], kinship = k_nona, start_snp1 = s1, n_snp = nsnp) -&gt; loglik
# in above call, s1 &amp; nsnp come from command line args
  calc_lrt_tib(loglik) -&gt; lrt[i]
}

fn_out &lt;- paste0("recla-boot-run", run_num, "_", proc_num, ".txt")
write.table(lrt, fn_out)
q("no")</code></pre>
</div>
<div id="files-in-data" class="section level3">
<h3 class="hasAnchor">
<a href="#files-in-data" class="anchor"></a>Files in <code>data</code>
</h3>
<p>The data subdirectory contains three rds files, one for each of the three input components.</p>
<ol style="list-style-type: decimal">
<li>founder allele dosages for Chr 8</li>
<li>phenotypes</li>
<li>kinship matrix (derived via LOCO method)</li>
</ol>
</div>
<div id="files-in-shell_scripts" class="section level3">
<h3 class="hasAnchor">
<a href="#files-in-shell_scripts" class="anchor"></a>Files in <code>shell_scripts</code>
</h3>
<p>We examine only the file that is needed for the Recla analysis, <code>boot-Recla-10-22.sh</code>.</p>
<p>Here is the contents of the file.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co">#!/bin/bash</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co"># untar your R installation</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="fu">tar</span> -xzf R13.tar.gz</a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="fu">tar</span> -xzf SLIBS.tar.gz</a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co"># make sure the script will use your R installation</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="bu">export</span> <span class="va">PATH=$(</span><span class="bu">pwd</span><span class="va">)</span>/R/bin:<span class="va">$PATH</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="bu">export</span> <span class="va">LD_LIBRARY_PATH=$(</span><span class="bu">pwd</span><span class="va">)</span>/SS:<span class="va">$LD_LIBRARY_PATH</span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co"># run R, with the name of your  R script</span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="ex">R</span> CMD BATCH <span class="st">'--args argname='</span><span class="va">$1</span><span class="st">' nsnp='</span><span class="va">$2</span><span class="st">' s1='</span><span class="va">$3</span><span class="st">' nboot_per_job='</span><span class="va">$4</span><span class="st">' run_num='</span><span class="va">$5</span><span class="st">''</span> Rscript/boot-Recla-10-22.R <span class="st">'boot400_run'</span><span class="va">$5</span><span class="st">'-'</span><span class="va">$1</span><span class="st">'.Rout'</span></a></code></pre></div>
<p>We need to unzip the R installation, which we’ve named R13.tar.gz, and SLIBS file on the remote computer where the actual computing will occur. Following that, and needed adjustments to the paths, we execute R. Note that we have, in this file, specified the variable values by numbers, like <code>$1</code> for <code>argname</code>. The values assigned to each of these is specified in the “submit” file.</p>
</div>
<div id="files-in-submit_files" class="section level3">
<h3 class="hasAnchor">
<a href="#files-in-submit_files" class="anchor"></a>Files in <code>submit_files</code>
</h3>
<p>The HTCondor “submit” file provides instructions for controlling the interaction with the high-throughput resources. Below is the text in the submit file <code>boot-Recla-10-22.sub</code></p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co"># hello-chtc.sub</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="co"># My very first HTCondor submit file</span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="va">s1=</span>650</a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co"># start scan at this SNP index value</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="va">nsnp=</span>350</a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="co"># length of scan in number of SNPs</span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co"># which run in the experimental design?</span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="va">run_num=</span>561</a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#</span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="va">nboot_per_job=</span>1</a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co">#  for almost all jobs), the desired name of the HTCondor log file,</span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#  and the desired name of the standard error file.  </span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"><span class="co">#  Wherever you see $(Cluster), HTCondor will insert the queue number</span></a>
<a class="sourceLine" id="cb4-14" data-line-number="14"><span class="co">#  assigned to this set of jobs at the time of submission.</span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="ex">universe</span> = vanilla</a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="ex">log</span> = boot-<span class="va">$(</span><span class="ex">Process</span><span class="va">)</span>-run<span class="va">$(</span><span class="ex">run_num</span><span class="va">)</span>.log</a>
<a class="sourceLine" id="cb4-17" data-line-number="17"><span class="ex">error</span> = boot-<span class="va">$(</span><span class="ex">Process</span><span class="va">)</span>-run<span class="va">$(</span><span class="ex">run_num</span><span class="va">)</span>.err</a>
<a class="sourceLine" id="cb4-18" data-line-number="18"><span class="co">#</span></a>
<a class="sourceLine" id="cb4-19" data-line-number="19"><span class="co"># Specify your executable (single binary or a script that runs several</span></a>
<a class="sourceLine" id="cb4-20" data-line-number="20"><span class="co">#  commands), arguments, and a files for HTCondor to store standard</span></a>
<a class="sourceLine" id="cb4-21" data-line-number="21"><span class="co">#  output (or "screen output").</span></a>
<a class="sourceLine" id="cb4-22" data-line-number="22"><span class="co">#  $(Process) will be a integer number for each job, starting with "0"</span></a>
<a class="sourceLine" id="cb4-23" data-line-number="23"><span class="co">#  and increasing for the relevant number of jobs.</span></a>
<a class="sourceLine" id="cb4-24" data-line-number="24"><span class="ex">executable</span> = ../shell_scripts/boot-Recla-10-22.sh</a>
<a class="sourceLine" id="cb4-25" data-line-number="25"><span class="ex">arguments</span> = <span class="va">$(</span><span class="ex">Process</span><span class="va">)</span> <span class="va">$(</span><span class="ex">nsnp</span><span class="va">)</span> <span class="va">$(</span><span class="ex">s1</span><span class="va">)</span> <span class="va">$(</span><span class="ex">nboot_per_job</span><span class="va">)</span> <span class="va">$(</span><span class="ex">run_num</span><span class="va">)</span></a>
<a class="sourceLine" id="cb4-26" data-line-number="26"><span class="ex">output</span> = boot-<span class="va">$(</span><span class="ex">Process</span><span class="va">)</span>-run<span class="va">$(</span><span class="ex">run_num</span><span class="va">)</span>.out</a>
<a class="sourceLine" id="cb4-27" data-line-number="27"><span class="co">#</span></a>
<a class="sourceLine" id="cb4-28" data-line-number="28"><span class="co"># Specify that HTCondor should transfer files to and from the</span></a>
<a class="sourceLine" id="cb4-29" data-line-number="29"><span class="co">#  computer where each job runs. The last of these lines *would* be</span></a>
<a class="sourceLine" id="cb4-30" data-line-number="30"><span class="ex">when_to_transfer_output</span> = ON_EXIT</a>
<a class="sourceLine" id="cb4-31" data-line-number="31"><span class="ex">transfer_input_files</span> = ../data,../Rscript,../shell_scripts,http://proxy.chtc.wisc.edu/SQUID/fjboehm/R13.tar.gz,http://proxy.chtc.wisc.edu/SQUID/SLIBS.tar.gz</a>
<a class="sourceLine" id="cb4-32" data-line-number="32"><span class="co">#</span></a>
<a class="sourceLine" id="cb4-33" data-line-number="33"><span class="co"># Tell HTCondor what amount of compute resources</span></a>
<a class="sourceLine" id="cb4-34" data-line-number="34"><span class="co">#  each job will need on the computer where it runs.</span></a>
<a class="sourceLine" id="cb4-35" data-line-number="35"><span class="ex">request_cpus</span> = 1</a>
<a class="sourceLine" id="cb4-36" data-line-number="36"><span class="ex">request_memory</span> = 3GB</a>
<a class="sourceLine" id="cb4-37" data-line-number="37"><span class="ex">request_disk</span> = 1GB</a>
<a class="sourceLine" id="cb4-38" data-line-number="38"><span class="co">#</span></a>
<a class="sourceLine" id="cb4-39" data-line-number="39"><span class="ex">requirements</span> = (OpSysMajorVer == 6) <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Logic">||</a></span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Paren">(</a></span><span class="ex">OpSysMajorVer</span> == 7<span class="kw">)</span></a>
<a class="sourceLine" id="cb4-40" data-line-number="40"></a>
<a class="sourceLine" id="cb4-41" data-line-number="41"><span class="co"># which computer grids to use:</span></a>
<a class="sourceLine" id="cb4-42" data-line-number="42"><span class="ex">+WantFlocking</span> = true</a>
<a class="sourceLine" id="cb4-43" data-line-number="43"><span class="ex">+WantGlideIn</span> = true</a>
<a class="sourceLine" id="cb4-44" data-line-number="44"></a>
<a class="sourceLine" id="cb4-45" data-line-number="45"><span class="co"># Tell HTCondor to run instances of our job:</span></a>
<a class="sourceLine" id="cb4-46" data-line-number="46"><span class="ex">queue</span> 1000</a></code></pre></div>
<p>The ordering of the arguments on in the line that starts with <code>arguments =</code> dictates the numbers that are assigned to each argument. These numbers are used in the shell script, above.</p>
</div>
<div id="files-on-squid" class="section level3">
<h3 class="hasAnchor">
<a href="#files-on-squid" class="anchor"></a>Files on <code>SQUID</code>
</h3>
<p>The UW-Madison CHTC provides for each user disk space on a web proxy. More information about <code>SQUID</code> can be found here: <a href="http://chtc.cs.wisc.edu/file-avail-squid.shtml" class="uri">http://chtc.cs.wisc.edu/file-avail-squid.shtml</a></p>
<p>In practice, we placed files containing founder allele dosages (for chromosomes of interest) and our (compressed) R installation on <code>SQUID</code>.</p>
</div>
</div>
<div id="submitting-jobs-to-condor-at-uw-madison-chtc" class="section level2">
<h2 class="hasAnchor">
<a href="#submitting-jobs-to-condor-at-uw-madison-chtc" class="anchor"></a>Submitting jobs to Condor (at UW-Madison CHTC)</h2>
<p>We followed the usual procedure for submitting R jobs with the CHTC. To submit the file <code>boot-Recla-10-22.sub</code>, we first changed into its directory. We then typed</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="ex">condor_submit</span> boot-Recla-10-22.sub</a></code></pre></div>
<p>By using this command from within the <code>submit_files</code> subdirectory, we ensure that our outputted files are returned to the <code>submit_files</code> subdirectory. We then manually moved the outputted results files to the <code>results</code> subdirectory.</p>
<p>We found that up to ten percent of jobs would initially fail and require re-submission. There are a variety of reasons that may explain the need for re-submission. Because of this need, we include in the git repository the file <code>boot-Recla-10-22-fix.sub</code>, which gives the code that we used for resubmission. The file <code>bad-jobs-run561</code> gives the ids of the jobs that failed the initial runs.</p>
</div>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">session_info</a></span>()</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="co">#&gt; ─ Session info ──────────────────────────────────────────────────────────</span></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="co">#&gt;  setting  value                       </span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co">#&gt;  version  R version 3.5.3 (2019-03-11)</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co">#&gt;  os       macOS Mojave 10.14.4        </span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">#&gt;  system   x86_64, darwin15.6.0        </span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#&gt;  ui       X11                         </span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="co">#&gt;  language (EN)                        </span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="co">#&gt;  collate  en_US.UTF-8                 </span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="co">#&gt;  ctype    en_US.UTF-8                 </span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="co">#&gt;  tz       America/Chicago             </span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="co">#&gt;  date     2019-04-28                  </span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="co">#&gt; ─ Packages ──────────────────────────────────────────────────────────────</span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="co">#&gt;  package     * version    date       lib source                        </span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16"><span class="co">#&gt;  assertthat    0.2.1      2019-03-21 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb6-17" data-line-number="17"><span class="co">#&gt;  backports     1.1.4      2019-04-10 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb6-18" data-line-number="18"><span class="co">#&gt;  callr         3.2.0      2019-03-15 [1] CRAN (R 3.5.3)                </span></a>
<a class="sourceLine" id="cb6-19" data-line-number="19"><span class="co">#&gt;  cli           1.1.0      2019-03-19 [1] CRAN (R 3.5.3)                </span></a>
<a class="sourceLine" id="cb6-20" data-line-number="20"><span class="co">#&gt;  commonmark    1.7        2018-12-01 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb6-21" data-line-number="21"><span class="co">#&gt;  crayon        1.3.4      2017-09-16 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb6-22" data-line-number="22"><span class="co">#&gt;  desc          1.2.0      2018-05-01 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb6-23" data-line-number="23"><span class="co">#&gt;  devtools      2.0.2      2019-04-08 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb6-24" data-line-number="24"><span class="co">#&gt;  digest        0.6.18     2018-10-10 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb6-25" data-line-number="25"><span class="co">#&gt;  evaluate      0.13       2019-02-12 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb6-26" data-line-number="26"><span class="co">#&gt;  formatR       1.6        2019-03-05 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb6-27" data-line-number="27"><span class="co">#&gt;  fs            1.2.7      2019-03-19 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb6-28" data-line-number="28"><span class="co">#&gt;  glue          1.3.1      2019-03-12 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb6-29" data-line-number="29"><span class="co">#&gt;  htmltools     0.3.6      2017-04-28 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb6-30" data-line-number="30"><span class="co">#&gt;  knitr         1.22       2019-03-08 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb6-31" data-line-number="31"><span class="co">#&gt;  magrittr      1.5        2014-11-22 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb6-32" data-line-number="32"><span class="co">#&gt;  MASS          7.3-51.1   2018-11-01 [1] CRAN (R 3.5.3)                </span></a>
<a class="sourceLine" id="cb6-33" data-line-number="33"><span class="co">#&gt;  memoise       1.1.0      2017-04-21 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb6-34" data-line-number="34"><span class="co">#&gt;  pkgbuild      1.0.3      2019-03-20 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb6-35" data-line-number="35"><span class="co">#&gt;  pkgdown       1.3.0.9000 2019-03-13 [1] Github (r-lib/pkgdown@0b2c27d)</span></a>
<a class="sourceLine" id="cb6-36" data-line-number="36"><span class="co">#&gt;  pkgload       1.0.2      2018-10-29 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb6-37" data-line-number="37"><span class="co">#&gt;  prettyunits   1.0.2      2015-07-13 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb6-38" data-line-number="38"><span class="co">#&gt;  processx      3.3.0      2019-03-10 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb6-39" data-line-number="39"><span class="co">#&gt;  ps            1.3.0      2018-12-21 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb6-40" data-line-number="40"><span class="co">#&gt;  R6            2.4.0      2019-02-14 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb6-41" data-line-number="41"><span class="co">#&gt;  Rcpp          1.0.1.2    2019-04-07 [1] Github (RcppCore/Rcpp@7f4e64d)</span></a>
<a class="sourceLine" id="cb6-42" data-line-number="42"><span class="co">#&gt;  remotes       2.0.4      2019-04-10 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb6-43" data-line-number="43"><span class="co">#&gt;  rlang         0.3.4      2019-04-07 [1] CRAN (R 3.5.3)                </span></a>
<a class="sourceLine" id="cb6-44" data-line-number="44"><span class="co">#&gt;  rmarkdown     1.11       2018-12-08 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb6-45" data-line-number="45"><span class="co">#&gt;  roxygen2      6.1.1      2018-11-07 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb6-46" data-line-number="46"><span class="co">#&gt;  rprojroot     1.3-2      2018-01-03 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb6-47" data-line-number="47"><span class="co">#&gt;  rstudioapi    0.10       2019-03-19 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb6-48" data-line-number="48"><span class="co">#&gt;  sessioninfo   1.1.1      2018-11-05 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb6-49" data-line-number="49"><span class="co">#&gt;  stringi       1.4.3      2019-03-12 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb6-50" data-line-number="50"><span class="co">#&gt;  stringr       1.4.0      2019-02-10 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb6-51" data-line-number="51"><span class="co">#&gt;  testthat      2.1.1      2019-04-23 [1] CRAN (R 3.5.3)                </span></a>
<a class="sourceLine" id="cb6-52" data-line-number="52"><span class="co">#&gt;  usethis       1.5.0      2019-04-07 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb6-53" data-line-number="53"><span class="co">#&gt;  withr         2.1.2      2018-03-15 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb6-54" data-line-number="54"><span class="co">#&gt;  xfun          0.5        2019-02-20 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb6-55" data-line-number="55"><span class="co">#&gt;  xml2          1.2.0      2018-01-24 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb6-56" data-line-number="56"><span class="co">#&gt;  yaml          2.2.0      2018-07-25 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb6-57" data-line-number="57"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb6-58" data-line-number="58"><span class="co">#&gt; [1] /Library/Frameworks/R.framework/Versions/3.5/Resources/library</span></a></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#overview">Overview</a></li>
      <li><a href="#directory-structure-files-setup-for-using-condor">Directory structure, files &amp; setup for using Condor</a></li>
      <li><a href="#submitting-jobs-to-condor-at-uw-madison-chtc">Submitting jobs to Condor (at UW-Madison CHTC)</a></li>
      <li><a href="#session-info">Session info</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Frederick Boehm.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9000.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
