<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Testing pleiotropy v separate QTL in multiparental populations • qtl2pleio</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Testing pleiotropy v separate QTL in multiparental populations">
<meta property="og:description" content="">
<meta property="og:image" content="https://fboehm.us/static/software/qtl2pleio/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">qtl2pleio</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.2.9001</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/bootstrap-cluster.html">Using HTCondor for bootstrap analysis</a>
    </li>
    <li>
      <a href="../articles/recla-analysis.html">Recla Analysis</a>
    </li>
    <li>
      <a href="../articles/testing-pleiotropy-vs-separate-qtl.html">Testing pleiotropy v separate QTL in multiparental populations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/fboehm/qtl2pleio">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Testing pleiotropy v separate QTL in multiparental populations</h1>
                        <h4 class="author">Frederick Boehm</h4>
            
            <h4 class="date">2019-03-16</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/fboehm/qtl2pleio/blob/master/vignettes/testing-pleiotropy-vs-separate-qtl.Rmd"><code>vignettes/testing-pleiotropy-vs-separate-qtl.Rmd</code></a></small>
      <div class="hidden name"><code>testing-pleiotropy-vs-separate-qtl.Rmd</code></div>

    </div>

    
    
<div id="setting-for-pleiotropy-vs-two-separate-qtl" class="section level2">
<h2 class="hasAnchor">
<a href="#setting-for-pleiotropy-vs-two-separate-qtl" class="anchor"></a>Setting for pleiotropy vs. two separate QTL</h2>
<p>Our setting involves a pair of traits, <span class="math inline">\(Y_1\)</span> and <span class="math inline">\(Y_2\)</span>, each of which individually (univariately) maps to a single genomic region. <span class="math inline">\(Y_1\)</span> and <span class="math inline">\(Y_2\)</span> are both measured on the same subjects. The exact definition of a genomic region is imprecise; in practice, it may be as large as 4 or 5 Mb. We seek to distinguish whether <span class="math inline">\(Y_1\)</span> and <span class="math inline">\(Y_2\)</span> associations (in the genomic region of interest) arise due to a single QTL or whether there are two two distinct loci, each of which associates with exactly one of the two traits. We recognize that more complicated association patterns are possible, but we neglect them in this test.</p>
</div>
<div id="installing-qtl2pleio" class="section level2">
<h2 class="hasAnchor">
<a href="#installing-qtl2pleio" class="anchor"></a>Installing <code>qtl2pleio</code>
</h2>
<p>We install <code>qtl2pleio</code> from github via the <code>devtools</code> R package, which is available from CRAN.</p>
<p>To install qtl2pleio, use <code>install_github()</code> from the <a href="https://devtools.r-lib.org">devtools</a> package.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/install.packages">install.packages</a></span>(<span class="st">"devtools"</span>)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"fboehm/qtl2pleio"</span>)</a></code></pre></div>
<p>You may also wish to install <a href="https://kbroman.org/qtl2">R/qtl2</a> and the <a href="https://github.com/rqtl/qtl2convert"><code>qtl2convert</code></a> package. We will use both below.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/install.packages">install.packages</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"qtl2"</span>, <span class="st">"qtl2convert"</span>), <span class="dt">repos =</span> <span class="st">"http://rqtl.org/qtl2"</span>)</a></code></pre></div>
<p>The above line only needs to be run once on a given computer (unless you wish to install a newer version of the package).</p>
<p>We then load the library into our R session with the <code>library</code> command:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(qtl2pleio)</a></code></pre></div>
<p>We’ll work with data from the <code>qtl2data</code> repository, which is on github. First, we install and load the <code>qtl2</code> package.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"rqtl/qtl2"</span>)</a></code></pre></div>
<p>We use the above line once to install the package on our computer before loading the package with the <code>library</code> command.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(qtl2)</a></code></pre></div>
</div>
<div id="reading-data-from-qtl2data-repository-on-github" class="section level2">
<h2 class="hasAnchor">
<a href="#reading-data-from-qtl2data-repository-on-github" class="anchor"></a>Reading data from <code>qtl2data</code> repository on github</h2>
<p>We’ll consider the <a href="https://github.com/rqtl/qtl2data/tree/master/DOex"><code>DOex</code></a> data in the <a href="https://github.com/rqtl/qtl2data"><code>qtl2data</code></a> repository. We’ll download the DOex.zip file before calculating founder allele dosages.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">file &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(<span class="st">"https://raw.githubusercontent.com/rqtl/"</span>, <span class="st">"qtl2data/master/DOex/DOex.zip"</span>)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">DOex &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/qtl2/topics/read_cross2">read_cross2</a></span>(file)</a></code></pre></div>
<p>Let’s calculate the founder allele dosages from the 36-state genotype probabilities.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">probs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/qtl2/topics/calc_genoprob">calc_genoprob</a></span>(DOex)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">pr &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/qtl2/topics/genoprob_to_alleleprob">genoprob_to_alleleprob</a></span>(probs)</a></code></pre></div>
<p>We now have an allele probabilities object stored in <code>pr</code>.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(pr)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="co">#&gt; [1] "2" "3" "X"</span></a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/dim">dim</a></span>(pr<span class="op">$</span><span class="st">`</span><span class="dt">3</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="co">#&gt; [1] 261   8 102</span></a></code></pre></div>
<p>We see that <code>pr</code> is a list of 3 three-dimensional arrays - one array for each of 3 chromosomes.</p>
<p>We now have an allele probabilities object stored in <code>pr</code>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(pr)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="co">#&gt; [1] "2" "3" "X"</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/dim">dim</a></span>(pr<span class="op">$</span><span class="st">`</span><span class="dt">3</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="co">#&gt; [1] 261   8 102</span></a></code></pre></div>
<p>We see that <code>pr</code> is a list of 3 three-dimensional arrays - one array for each of 3 chromosomes.</p>
</div>
<div id="kinship-calculations" class="section level2">
<h2 class="hasAnchor">
<a href="#kinship-calculations" class="anchor"></a>Kinship calculations</h2>
<p>For our statistical model, we need a kinship matrix. Although we don’t have genome-wide data - since we have allele probabilities for only 3 chromosomes - let’s calculate a kinship matrix using “leave-one-chromosome-out”. In practice, one would want to use allele probabilities from a full genome-wide set of markers.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">kinship &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/qtl2/topics/calc_kinship">calc_kinship</a></span>(<span class="dt">probs =</span> pr, <span class="dt">type =</span> <span class="st">"loco"</span>)</a></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/str">str</a></span>(kinship)</a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="co">#&gt; List of 3</span></a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="co">#&gt;  $ 2: num [1:261, 1:261] 0.6934 0.0705 0.2356 0.0558 0.0513 ...</span></a>
<a class="sourceLine" id="cb11-4" data-line-number="4"><span class="co">#&gt;   ..- attr(*, "n_pos")= int 195</span></a>
<a class="sourceLine" id="cb11-5" data-line-number="5"><span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span></a>
<a class="sourceLine" id="cb11-6" data-line-number="6"><span class="co">#&gt;   .. ..$ : chr [1:261] "1" "4" "5" "6" ...</span></a>
<a class="sourceLine" id="cb11-7" data-line-number="7"><span class="co">#&gt;   .. ..$ : chr [1:261] "1" "4" "5" "6" ...</span></a>
<a class="sourceLine" id="cb11-8" data-line-number="8"><span class="co">#&gt;  $ 3: num [1:261, 1:261] 0.6662 0.0647 0.2024 0.1129 0.0772 ...</span></a>
<a class="sourceLine" id="cb11-9" data-line-number="9"><span class="co">#&gt;   ..- attr(*, "n_pos")= int 220</span></a>
<a class="sourceLine" id="cb11-10" data-line-number="10"><span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span></a>
<a class="sourceLine" id="cb11-11" data-line-number="11"><span class="co">#&gt;   .. ..$ : chr [1:261] "1" "4" "5" "6" ...</span></a>
<a class="sourceLine" id="cb11-12" data-line-number="12"><span class="co">#&gt;   .. ..$ : chr [1:261] "1" "4" "5" "6" ...</span></a>
<a class="sourceLine" id="cb11-13" data-line-number="13"><span class="co">#&gt;  $ X: num [1:261, 1:261] 0.4871 0.0831 0.1953 0.1043 0.1125 ...</span></a>
<a class="sourceLine" id="cb11-14" data-line-number="14"><span class="co">#&gt;   ..- attr(*, "n_pos")= int 229</span></a>
<a class="sourceLine" id="cb11-15" data-line-number="15"><span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span></a>
<a class="sourceLine" id="cb11-16" data-line-number="16"><span class="co">#&gt;   .. ..$ : chr [1:261] "1" "4" "5" "6" ...</span></a>
<a class="sourceLine" id="cb11-17" data-line-number="17"><span class="co">#&gt;   .. ..$ : chr [1:261] "1" "4" "5" "6" ...</span></a></code></pre></div>
<p>We see that <code>kinship</code> is a list containing 3 matrices. Each matrix is 261 by 261 - where the number of subjects is 261 - and symmetric.</p>
</div>
<div id="statistical-model" class="section level2">
<h2 class="hasAnchor">
<a href="#statistical-model" class="anchor"></a>Statistical model</h2>
<p>Before we simulate phenotype data, we first specify our statistical model.</p>
<p>We use the model:</p>
<p><span class="math display">\[vec(Y) = X vec(B) + vec(G) + vec(E)\]</span></p>
<p>where <span class="math inline">\(Y\)</span> is a <span class="math inline">\(n\)</span> by <span class="math inline">\(2\)</span> matrix, where each row is one subject and each column is one quantitative trait. <span class="math inline">\(X\)</span> is a <span class="math inline">\(2n\)</span> by <span class="math inline">\(2f\)</span> design matrix containing <span class="math inline">\(n\)</span> by <span class="math inline">\(f\)</span> allele probabilities matrices for each of two (possibly identical) markers. Thus, <span class="math inline">\(X\)</span> is a block-diagonal matrix, with exactly two <span class="math inline">\(n\)</span> by <span class="math inline">\(f\)</span> blocks on the diagonal. <span class="math inline">\(B\)</span> is a <span class="math inline">\(f\)</span> by 2 matrix. “vec” refers to the vectorization operator. “vec(B)”, where <span class="math inline">\(B\)</span> is a <span class="math inline">\(f\)</span> by <span class="math inline">\(2\)</span> matrix, is, thus, a (column) vector of length <span class="math inline">\(2f\)</span> that is formed by stacking the second column of <span class="math inline">\(B\)</span> beneath the first column of <span class="math inline">\(B\)</span>.</p>
<p><span class="math inline">\(G\)</span> is a matrix of random effects. We specify its distribution as matrix-variate normal with mean being a <span class="math inline">\(n\)</span> by <span class="math inline">\(2\)</span> matrix of zeros, covariance among row vectors a <span class="math inline">\(n\)</span> by <span class="math inline">\(n\)</span> kinship matrix, <span class="math inline">\(K\)</span>, and covariance among column vectors a <span class="math inline">\(2\)</span> by <span class="math inline">\(2\)</span> genetic covariance matrix, <span class="math inline">\(V_g\)</span>.</p>
<p>In mathematical notation, we write:</p>
<p><span class="math display">\[G \sim MN_{\text{n by 2}}(0, K, V_g)\]</span></p>
<p>We also need to specify the distribution of the <span class="math inline">\(E\)</span> matrix, which contains the random errors. <span class="math inline">\(E\)</span> is a random <span class="math inline">\(n\)</span> by <span class="math inline">\(2\)</span> matrix that is distributed as a matrix-variate normal distribution with mean being the <span class="math inline">\(n\)</span> by <span class="math inline">\(2\)</span> zero matrix, covariance among row vectors <span class="math inline">\(I_n\)</span>, the <span class="math inline">\(n\)</span> by <span class="math inline">\(n\)</span> identity matrix, and covariance among columns the <span class="math inline">\(2\)</span> by <span class="math inline">\(2\)</span> matrix <span class="math inline">\(V_e\)</span>.</p>
<p><span class="math display">\[E \sim MN_{\text{n by 2}}(0, I_n, V_e)\]</span></p>
<p>In practice, we typically measure the phenotype matrix <span class="math inline">\(Y\)</span>. We also treat as known the design matrix <span class="math inline">\(X\)</span> and the kinship matrix <span class="math inline">\(K\)</span>. We then infer the values of <span class="math inline">\(B\)</span>, <span class="math inline">\(V_g\)</span>, and <span class="math inline">\(V_e\)</span>.</p>
</div>
<div id="simulating-phenotypes-with-qtl2pleiosim1" class="section level2">
<h2 class="hasAnchor">
<a href="#simulating-phenotypes-with-qtl2pleiosim1" class="anchor"></a>Simulating phenotypes with <code><a href="../reference/sim1.html">qtl2pleio::sim1</a></code>
</h2>
<p>The function to simulate phenotypes in <code>qtl2pleio</code> is <code>sim1</code>. By examining its help page, we see that it takes five arguments. The help page also gives the dimensions of the inputs.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="co"># set up the design matrix, X</span></a>
<a class="sourceLine" id="cb12-2" data-line-number="2">pp &lt;-<span class="st"> </span>pr[[<span class="dv">2</span>]]  <span class="co">#we'll work with Chr 3's genotype data</span></a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/dim">dim</a></span>(pp)</a>
<a class="sourceLine" id="cb12-4" data-line-number="4"><span class="co">#&gt; [1] 261   8 102</span></a></code></pre></div>
<p>We prepare a block-diagonal design matrix X that contains two nonzero blocks on the diagonal, one for each trait. We use here a function from the <code>gemma2</code> R package to set up the needed matrix.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="co"># Next, we prepare a design matrix X</span></a>
<a class="sourceLine" id="cb13-2" data-line-number="2">X &lt;-<span class="st"> </span>gemma2<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/gemma2/topics/stagger_mats">stagger_mats</a></span>(pp[, , <span class="dv">50</span>], pp[, , <span class="dv">50</span>])</a>
<a class="sourceLine" id="cb13-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/dim">dim</a></span>(X)</a>
<a class="sourceLine" id="cb13-4" data-line-number="4"><span class="co">#&gt; [1] 522  16</span></a></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="co"># assemble B matrix of allele effects</span></a>
<a class="sourceLine" id="cb14-2" data-line-number="2">B &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">matrix</a></span>(<span class="dt">data =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="op">-</span><span class="dv">1</span>, <span class="dv">-1</span>, <span class="dv">-1</span>, <span class="dv">-1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">-1</span>, <span class="dv">-1</span>, <span class="dv">-1</span>, </a>
<a class="sourceLine" id="cb14-3" data-line-number="3">    <span class="dv">-1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>), <span class="dt">nrow =</span> <span class="dv">8</span>, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">byrow =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="co"># verify that B is what we want:</span></a>
<a class="sourceLine" id="cb14-5" data-line-number="5">B</a>
<a class="sourceLine" id="cb14-6" data-line-number="6"><span class="co">#&gt;      [,1] [,2]</span></a>
<a class="sourceLine" id="cb14-7" data-line-number="7"><span class="co">#&gt; [1,]   -1   -1</span></a>
<a class="sourceLine" id="cb14-8" data-line-number="8"><span class="co">#&gt; [2,]   -1   -1</span></a>
<a class="sourceLine" id="cb14-9" data-line-number="9"><span class="co">#&gt; [3,]   -1   -1</span></a>
<a class="sourceLine" id="cb14-10" data-line-number="10"><span class="co">#&gt; [4,]   -1   -1</span></a>
<a class="sourceLine" id="cb14-11" data-line-number="11"><span class="co">#&gt; [5,]    1    1</span></a>
<a class="sourceLine" id="cb14-12" data-line-number="12"><span class="co">#&gt; [6,]    1    1</span></a>
<a class="sourceLine" id="cb14-13" data-line-number="13"><span class="co">#&gt; [7,]    1    1</span></a>
<a class="sourceLine" id="cb14-14" data-line-number="14"><span class="co">#&gt; [8,]    1    1</span></a>
<a class="sourceLine" id="cb14-15" data-line-number="15"><span class="co"># set.seed to ensure reproducibility</span></a>
<a class="sourceLine" id="cb14-16" data-line-number="16"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">2018</span> <span class="op">-</span><span class="st"> </span><span class="dv">1</span> <span class="op">-</span><span class="st"> </span><span class="dv">30</span>)</a>
<a class="sourceLine" id="cb14-17" data-line-number="17"><span class="co"># call to sim1</span></a>
<a class="sourceLine" id="cb14-18" data-line-number="18">Ypre &lt;-<span class="st"> </span><span class="kw"><a href="../reference/sim1.html">sim1</a></span>(<span class="dt">X =</span> X, <span class="dt">B =</span> B, <span class="dt">Vg =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/diag">diag</a></span>(<span class="dv">2</span>), <span class="dt">Ve =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/diag">diag</a></span>(<span class="dv">2</span>), <span class="dt">kinship =</span> kinship[[<span class="dv">2</span>]])</a>
<a class="sourceLine" id="cb14-19" data-line-number="19">Y &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">matrix</a></span>(Ypre, <span class="dt">nrow =</span> <span class="dv">261</span>, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">byrow =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb14-20" data-line-number="20"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(Y) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(pp)</a>
<a class="sourceLine" id="cb14-21" data-line-number="21"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(Y) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"tr1"</span>, <span class="st">"tr2"</span>)</a></code></pre></div>
<p>Let’s perform univariate QTL mapping for each of the two traits in the Y matrix.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">s1 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/qtl2/topics/scan1">scan1</a></span>(<span class="dt">genoprobs =</span> pr, <span class="dt">pheno =</span> Y, <span class="dt">kinship =</span> kinship)</a></code></pre></div>
<p>Here is a plot of the results.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(s1, DOex<span class="op">$</span>pmap<span class="op">$</span><span class="st">`</span><span class="dt">3</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(s1, DOex<span class="op">$</span>pmap<span class="op">$</span><span class="st">`</span><span class="dt">3</span><span class="st">`</span>, <span class="dt">lod =</span> <span class="dv">2</span>, <span class="dt">col =</span> <span class="st">"violetred"</span>, <span class="dt">add =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb16-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/legend">legend</a></span>(<span class="st">"topleft"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(s1), <span class="dt">lwd =</span> <span class="dv">2</span>, <span class="dt">col =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"darkslateblue"</span>, </a>
<a class="sourceLine" id="cb16-4" data-line-number="4">    <span class="st">"violetred"</span>), <span class="dt">bg =</span> <span class="st">"gray92"</span>)</a></code></pre></div>
<p><img src="testing-pleiotropy-vs-separate-qtl_files/figure-html/1d-lod-plots-1.png" width="700"></p>
<p>We see that the two traits share a peak on Chr 3.</p>
<p>And here are the observed QTL peaks with LOD &gt; 8. In practice, we could do a permutation test to determine a threshold for family-wise error rate control.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/qtl2/topics/find_peaks">find_peaks</a></span>(s1, <span class="dt">map =</span> DOex<span class="op">$</span>pmap, <span class="dt">threshold =</span> <span class="dv">8</span>)</a>
<a class="sourceLine" id="cb17-2" data-line-number="2"><span class="co">#&gt;   lodindex lodcolumn chr       pos      lod</span></a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="co">#&gt; 1        1       tr1   3  82.77806 16.19704</span></a>
<a class="sourceLine" id="cb17-4" data-line-number="4"><span class="co">#&gt; 2        2       tr2   3  82.77806 18.26406</span></a>
<a class="sourceLine" id="cb17-5" data-line-number="5"><span class="co">#&gt; 3        2       tr2   X 103.79061 16.19708</span></a></code></pre></div>
</div>
<div id="perform-two-dimensional-scan-as-first-step-in-pleiotropy-v-separate-qtl-hypothesis-test" class="section level1">
<h1 class="hasAnchor">
<a href="#perform-two-dimensional-scan-as-first-step-in-pleiotropy-v-separate-qtl-hypothesis-test" class="anchor"></a>Perform two-dimensional scan as first step in pleiotropy v separate QTL hypothesis test</h1>
<p>We now have the inputs that we need to do a pleiotropy vs. separate QTL test. We have the founder allele dosages for one chromosome, <em>i.e.</em>, Chr 3, in the R object <code>pp</code>, the matrix of two trait measurements in <code>Y</code>, and a LOCO-derived kinship matrix. We also specify, via the <code>start_snp</code> argument, the starting point for the two-dimensional scan within the array of founder allele dosages. Here, we choose the 38th marker in the array as the starting point. Via the <code>n_snp</code> argument, we specify the number of markers to include in the two-dimensional scan. Here, we input 25, so that we fit the bivariate linear mixed effects model at 25*25 = 625 ordered pairs of markers. In practice, we usually use between 100 and 300 markers for most two-dimensional scans.</p>
<p>Lastly, we specify the number of cores to use, with the <code>n_cores</code> argument. We set it to 1 here, to ensure that the vignette can be run by CRAN. However, in practice, you may wish to increase the number of cores to accelerate computing.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">out &lt;-<span class="st"> </span><span class="kw"><a href="../reference/scan_pvl.html">scan_pvl</a></span>(<span class="dt">probs =</span> pp, <span class="dt">pheno =</span> Y, <span class="dt">kinship =</span> kinship<span class="op">$</span><span class="st">`</span><span class="dt">3</span><span class="st">`</span>, </a>
<a class="sourceLine" id="cb18-2" data-line-number="2">    <span class="dt">start_snp =</span> <span class="dv">38</span>, <span class="dt">n_snp =</span> <span class="dv">25</span>, <span class="dt">n_cores =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb18-3" data-line-number="3"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb18-4" data-line-number="4"><span class="co">#&gt; covariance matrices estimation completed.</span></a></code></pre></div>
<p>The number of cores available will vary by computer. For example, on my Macbook pro computer, with 16GB RAM, I have access to 8 cores. If I use all 8, I can’t do other computing tasks, so I often set <code>n_cores</code> to 7.</p>
<p>To check how many cores are available on your computer, run this code.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">parallel<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/parallel/topics/detectCores">detectCores</a></span>()</a></code></pre></div>
<div id="create-a-profile-lod-plot-to-visualize-results-of-two-dimensional-scan" class="section level4">
<h4 class="hasAnchor">
<a href="#create-a-profile-lod-plot-to-visualize-results-of-two-dimensional-scan" class="anchor"></a>Create a profile LOD plot to visualize results of two-dimensional scan</h4>
<p>To visualize results from our two-dimensional scan, we calculate profile LOD for each trait. The code below makes use of the R package <code>ggplot2</code> to plot profile LODs over the scan region.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">out</a>
<a class="sourceLine" id="cb20-2" data-line-number="2"><span class="co">#&gt; # A tibble: 625 x 3</span></a>
<a class="sourceLine" id="cb20-3" data-line-number="3"><span class="co">#&gt;    Var1               Var2        loglik</span></a>
<a class="sourceLine" id="cb20-4" data-line-number="4"><span class="co">#&gt;    &lt;chr&gt;              &lt;chr&gt;        &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb20-5" data-line-number="5"><span class="co">#&gt;  1 JAX00108034        JAX00108034  -866.</span></a>
<a class="sourceLine" id="cb20-6" data-line-number="6"><span class="co">#&gt;  2 backupUNC031096286 JAX00108034  -863.</span></a>
<a class="sourceLine" id="cb20-7" data-line-number="7"><span class="co">#&gt;  3 JAX00525579        JAX00108034  -865.</span></a>
<a class="sourceLine" id="cb20-8" data-line-number="8"><span class="co">#&gt;  4 JAX00525718        JAX00108034  -865.</span></a>
<a class="sourceLine" id="cb20-9" data-line-number="9"><span class="co">#&gt;  5 UNC030088171       JAX00108034  -864.</span></a>
<a class="sourceLine" id="cb20-10" data-line-number="10"><span class="co">#&gt;  6 backupUNC030474070 JAX00108034  -864.</span></a>
<a class="sourceLine" id="cb20-11" data-line-number="11"><span class="co">#&gt;  7 backupUNC030474244 JAX00108034  -863.</span></a>
<a class="sourceLine" id="cb20-12" data-line-number="12"><span class="co">#&gt;  8 UNC030103315       JAX00108034  -862.</span></a>
<a class="sourceLine" id="cb20-13" data-line-number="13"><span class="co">#&gt;  9 UNC030107226       JAX00108034  -861.</span></a>
<a class="sourceLine" id="cb20-14" data-line-number="14"><span class="co">#&gt; 10 JAX00527615        JAX00108034  -861.</span></a>
<a class="sourceLine" id="cb20-15" data-line-number="15"><span class="co">#&gt; # … with 615 more rows</span></a></code></pre></div>
<p>We see that <code>out</code> is a 625 by 3 tibble, as expected. The first two columns contain the marker ids for each ordered pair of markers. The third column contains the log-likelihood values.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr)</a>
<a class="sourceLine" id="cb21-2" data-line-number="2"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-3" data-line-number="3"><span class="co">#&gt; Attaching package: 'dplyr'</span></a>
<a class="sourceLine" id="cb21-4" data-line-number="4"><span class="co">#&gt; The following objects are masked from 'package:stats':</span></a>
<a class="sourceLine" id="cb21-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-6" data-line-number="6"><span class="co">#&gt;     filter, lag</span></a>
<a class="sourceLine" id="cb21-7" data-line-number="7"><span class="co">#&gt; The following objects are masked from 'package:base':</span></a>
<a class="sourceLine" id="cb21-8" data-line-number="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb21-9" data-line-number="9"><span class="co">#&gt;     intersect, setdiff, setequal, union</span></a>
<a class="sourceLine" id="cb21-10" data-line-number="10">out <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="../reference/tidy_scan_pvl.html">tidy_scan_pvl</a></span>(DOex<span class="op">$</span>pmap<span class="op">$</span><span class="st">`</span><span class="dt">3</span><span class="st">`</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="../reference/add_intercepts.html">add_intercepts</a></span>(<span class="dt">intercepts_univariate =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="fl">82.8</span>, </a>
<a class="sourceLine" id="cb21-11" data-line-number="11">    <span class="fl">82.8</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="../reference/plot_pvl.html">plot_pvl</a></span>(<span class="dt">phenames =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"tr1"</span>, <span class="st">"tr2"</span>))</a>
<a class="sourceLine" id="cb21-12" data-line-number="12"><span class="co">#&gt; Warning: Removed 49 rows containing missing values (geom_path).</span></a></code></pre></div>
<p><img src="testing-pleiotropy-vs-separate-qtl_files/figure-html/profile-plot-1.png" width="700"></p>
<p>We first pass the <code>scan_pvl</code> output, , <code>out</code>, to the function <code>tidy_scan_pvl</code> to add the physical map coordinates to the <code>out</code> tibble. We pipe that output to the <code>add_intercepts</code> function. This function adds columns for the univariate peak positions. Note that we need to specify the univariate peak positions by hand. In the current case, the two traits have identical peak positions.</p>
<p>Finally, the output of <code>add_intercepts</code> is piped to <code>plot_pvl</code>. This function uses <code>ggplot2</code> functions to create a profile LOD plot with three “traces”: one for each trait and a third for all ordered pairs under the pleiotropy hypothesis.</p>
</div>
<div id="calculate-the-likelihood-ratio-test-statistic-for-pleiotropy-v-separate-qtl" class="section level4">
<h4 class="hasAnchor">
<a href="#calculate-the-likelihood-ratio-test-statistic-for-pleiotropy-v-separate-qtl" class="anchor"></a>Calculate the likelihood ratio test statistic for pleiotropy v separate QTL</h4>
<p>We use the function <code>calc_lrt_tib</code> to calculate the likelihood ratio test statistic value for the specified traits and specified genomic region.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1">(lrt &lt;-<span class="st"> </span><span class="kw"><a href="../reference/calc_lrt_tib.html">calc_lrt_tib</a></span>(out))</a>
<a class="sourceLine" id="cb22-2" data-line-number="2"><span class="co">#&gt; [1] 0</span></a></code></pre></div>
</div>
<div id="bootstrap-analysis-to-get-p-values" class="section level3">
<h3 class="hasAnchor">
<a href="#bootstrap-analysis-to-get-p-values" class="anchor"></a>Bootstrap analysis to get p-values</h3>
<p>The calibration of test statistic values to get p-values uses bootstrap methods because we don’t know the theoretical distribution of the test statistic under the null hypothesis. Thus, we use a bootstrap approach to obtain an empirical distribution of test statistic values under the null hypothesis of the presence of one pleiotropic locus.</p>
<p>We will use the function <code>boot_pvl</code> from our package <code>qtl2pleio</code>.</p>
<p>We use a parametric bootstrap strategy in which we first use the studied phenotypes to infer the values of model parameters. Once we have the inferred values of the model parameters, we simulate phenotypes from the pleiotropy model (with the inferred parameter values).</p>
<p>A natural question that arises is “which marker’s allele probabilities do we use when simulating phenotypes?” We use the marker that, under the null hypothesis, <em>i.e.</em>, under the pleiotropy constraint, yields the greatest value of the log-likelihood.</p>
<p>Before we call <code>boot_pvl</code>, we need to identify the index (on the chromosome under study) of the marker that maximizes the likelihood under the pleiotropy constraint. To do this, we use the <code>qtl2pleio</code> function <code>find_pleio_peak_tib</code>.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">(pleio_index &lt;-<span class="st"> </span><span class="kw"><a href="../reference/find_pleio_peak_tib.html">find_pleio_peak_tib</a></span>(out, <span class="dt">start_snp =</span> <span class="dv">38</span>))</a>
<a class="sourceLine" id="cb23-2" data-line-number="2"><span class="co">#&gt; loglik13 </span></a>
<a class="sourceLine" id="cb23-3" data-line-number="3"><span class="co">#&gt;       50</span></a></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">2018</span> <span class="op">-</span><span class="st"> </span><span class="dv">11</span> <span class="op">-</span><span class="st"> </span><span class="dv">25</span>)</a>
<a class="sourceLine" id="cb24-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.time">system.time</a></span>(b_out &lt;-<span class="st"> </span><span class="kw"><a href="../reference/boot_pvl.html">boot_pvl</a></span>(<span class="dt">probs =</span> pp, <span class="dt">pheno =</span> Y, <span class="dt">pleio_peak_index =</span> pleio_index, </a>
<a class="sourceLine" id="cb24-3" data-line-number="3">    <span class="dt">kinship =</span> kinship<span class="op">$</span><span class="st">`</span><span class="dt">3</span><span class="st">`</span>, <span class="dt">nboot_per_job =</span> <span class="dv">10</span>, <span class="dt">start_snp =</span> <span class="dv">38</span>, </a>
<a class="sourceLine" id="cb24-4" data-line-number="4">    <span class="dt">n_snp =</span> <span class="dv">25</span>))</a>
<a class="sourceLine" id="cb24-5" data-line-number="5"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb24-6" data-line-number="6"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb24-7" data-line-number="7"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb24-8" data-line-number="8"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb24-9" data-line-number="9"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb24-10" data-line-number="10"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb24-11" data-line-number="11"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb24-12" data-line-number="12"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb24-13" data-line-number="13"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb24-14" data-line-number="14"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb24-15" data-line-number="15"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb24-16" data-line-number="16"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb24-17" data-line-number="17"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb24-18" data-line-number="18"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb24-19" data-line-number="19"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb24-20" data-line-number="20"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb24-21" data-line-number="21"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb24-22" data-line-number="22"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb24-23" data-line-number="23"><span class="co">#&gt; starting covariance matrices estimation with data from 261 subjects.</span></a>
<a class="sourceLine" id="cb24-24" data-line-number="24"><span class="co">#&gt; covariance matrices estimation completed.</span></a>
<a class="sourceLine" id="cb24-25" data-line-number="25"><span class="co">#&gt;    user  system elapsed </span></a>
<a class="sourceLine" id="cb24-26" data-line-number="26"><span class="co">#&gt;  79.487   1.629  81.350</span></a></code></pre></div>
<p>The argument <code>nboot_per_job</code> indicates the number of bootstrap samples that will be created and analyzed. Here, we set <code>nboot_per_job = 10</code>, so we expect to see returned a numeric vector of length 10, where each entry is a LRT statistic value from a distinct bootstrap sample.</p>
<p>Finally, we determine a bootstrap p-value in the usual method. We treat the bootstrap samples’ test statistics as an empirical distribution of the test statistic under the null hypothesis of pleiotropy. Thus, to get a p-value, we want to ask “What is the probability, under the null hypothesis, of observing a test statistic value that is at least as extreme as that which we observed?”</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1">b_out</a>
<a class="sourceLine" id="cb25-2" data-line-number="2"><span class="co">#&gt;  [1] 2.8881948 0.0000000 0.9943472 0.0000000 1.1553322 0.6764714 0.0000000</span></a>
<a class="sourceLine" id="cb25-3" data-line-number="3"><span class="co">#&gt;  [8] 1.1446433 0.0000000 1.0460109</span></a>
<a class="sourceLine" id="cb25-4" data-line-number="4">(pvalue &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/mean">mean</a></span>(b_out <span class="op">&gt;=</span><span class="st"> </span>lrt))</a>
<a class="sourceLine" id="cb25-5" data-line-number="5"><span class="co">#&gt; [1] 1</span></a></code></pre></div>
<p>In practice, one would want to use many more bootstrap samples to achieve an empirical distribution that is closer to the theoretical distribution of the test statistic under the null hypothesis.</p>
<p>However, if one wants to perform analyses with a reasonable number - say 400 - bootstrap samples, this will take a very long time - many days - on a single laptop computer. We have used a series of computer clusters that are coordinated by the University of Wisconsin-Madison’s Center for High-throughput Computing (<a href="http://chtc.cs.wisc.edu" class="uri">http://chtc.cs.wisc.edu</a>). We typically are able to analyze 1000 bootstrap samples in less than 24 hours with this service.</p>
</div>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">session_info</a></span>()</a>
<a class="sourceLine" id="cb26-2" data-line-number="2"><span class="co">#&gt; ─ Session info ──────────────────────────────────────────────────────────</span></a>
<a class="sourceLine" id="cb26-3" data-line-number="3"><span class="co">#&gt;  setting  value                                      </span></a>
<a class="sourceLine" id="cb26-4" data-line-number="4"><span class="co">#&gt;  version  R version 3.5.2 Patched (2018-12-24 r75893)</span></a>
<a class="sourceLine" id="cb26-5" data-line-number="5"><span class="co">#&gt;  os       macOS Mojave 10.14.3                       </span></a>
<a class="sourceLine" id="cb26-6" data-line-number="6"><span class="co">#&gt;  system   x86_64, darwin15.6.0                       </span></a>
<a class="sourceLine" id="cb26-7" data-line-number="7"><span class="co">#&gt;  ui       X11                                        </span></a>
<a class="sourceLine" id="cb26-8" data-line-number="8"><span class="co">#&gt;  language (EN)                                       </span></a>
<a class="sourceLine" id="cb26-9" data-line-number="9"><span class="co">#&gt;  collate  en_US.UTF-8                                </span></a>
<a class="sourceLine" id="cb26-10" data-line-number="10"><span class="co">#&gt;  ctype    en_US.UTF-8                                </span></a>
<a class="sourceLine" id="cb26-11" data-line-number="11"><span class="co">#&gt;  tz       America/Chicago                            </span></a>
<a class="sourceLine" id="cb26-12" data-line-number="12"><span class="co">#&gt;  date     2019-03-16                                 </span></a>
<a class="sourceLine" id="cb26-13" data-line-number="13"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb26-14" data-line-number="14"><span class="co">#&gt; ─ Packages ──────────────────────────────────────────────────────────────</span></a>
<a class="sourceLine" id="cb26-15" data-line-number="15"><span class="co">#&gt;  package     * version    date       lib source                        </span></a>
<a class="sourceLine" id="cb26-16" data-line-number="16"><span class="co">#&gt;  assertthat    0.2.0      2017-04-11 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-17" data-line-number="17"><span class="co">#&gt;  backports     1.1.3      2018-12-14 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-18" data-line-number="18"><span class="co">#&gt;  bit           1.1-14     2018-05-29 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-19" data-line-number="19"><span class="co">#&gt;  bit64         0.9-7      2017-05-08 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-20" data-line-number="20"><span class="co">#&gt;  blob          1.1.1      2018-03-25 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-21" data-line-number="21"><span class="co">#&gt;  callr         3.1.1      2018-12-21 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-22" data-line-number="22"><span class="co">#&gt;  cli           1.0.1      2018-09-25 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-23" data-line-number="23"><span class="co">#&gt;  colorspace    1.4-0      2019-01-13 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-24" data-line-number="24"><span class="co">#&gt;  commonmark    1.7        2018-12-01 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-25" data-line-number="25"><span class="co">#&gt;  crayon        1.3.4      2017-09-16 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-26" data-line-number="26"><span class="co">#&gt;  data.table    1.12.0     2019-01-13 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-27" data-line-number="27"><span class="co">#&gt;  DBI           1.0.0      2018-05-02 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-28" data-line-number="28"><span class="co">#&gt;  desc          1.2.0      2018-05-01 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-29" data-line-number="29"><span class="co">#&gt;  devtools      2.0.1      2018-10-26 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-30" data-line-number="30"><span class="co">#&gt;  digest        0.6.18     2018-10-10 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-31" data-line-number="31"><span class="co">#&gt;  dplyr       * 0.8.0.1    2019-02-15 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-32" data-line-number="32"><span class="co">#&gt;  evaluate      0.13       2019-02-12 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-33" data-line-number="33"><span class="co">#&gt;  fansi         0.4.0      2018-10-05 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-34" data-line-number="34"><span class="co">#&gt;  formatR       1.6        2019-03-05 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-35" data-line-number="35"><span class="co">#&gt;  fs            1.2.6      2018-08-23 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-36" data-line-number="36"><span class="co">#&gt;  gemma2        0.0.1.1    2019-03-13 [1] Github (fboehm/gemma2@25fb644)</span></a>
<a class="sourceLine" id="cb26-37" data-line-number="37"><span class="co">#&gt;  ggplot2       3.1.0      2018-10-25 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-38" data-line-number="38"><span class="co">#&gt;  glue          1.3.1      2019-03-12 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-39" data-line-number="39"><span class="co">#&gt;  gtable        0.2.0      2016-02-26 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-40" data-line-number="40"><span class="co">#&gt;  htmltools     0.3.6      2017-04-28 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-41" data-line-number="41"><span class="co">#&gt;  jsonlite      1.6        2018-12-07 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-42" data-line-number="42"><span class="co">#&gt;  knitr         1.22       2019-03-08 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-43" data-line-number="43"><span class="co">#&gt;  labeling      0.3        2014-08-23 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-44" data-line-number="44"><span class="co">#&gt;  lattice       0.20-38    2018-11-04 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-45" data-line-number="45"><span class="co">#&gt;  lazyeval      0.2.1      2017-10-29 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-46" data-line-number="46"><span class="co">#&gt;  magrittr      1.5        2014-11-22 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-47" data-line-number="47"><span class="co">#&gt;  MASS          7.3-51.1   2018-11-01 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-48" data-line-number="48"><span class="co">#&gt;  Matrix        1.2-16     2019-03-08 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-49" data-line-number="49"><span class="co">#&gt;  memoise       1.1.0      2017-04-21 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-50" data-line-number="50"><span class="co">#&gt;  munsell       0.5.0      2018-06-12 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-51" data-line-number="51"><span class="co">#&gt;  pillar        1.3.1      2018-12-15 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-52" data-line-number="52"><span class="co">#&gt;  pkgbuild      1.0.2      2018-10-16 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-53" data-line-number="53"><span class="co">#&gt;  pkgconfig     2.0.2      2018-08-16 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-54" data-line-number="54"><span class="co">#&gt;  pkgdown       1.3.0.9000 2019-03-13 [1] Github (r-lib/pkgdown@0b2c27d)</span></a>
<a class="sourceLine" id="cb26-55" data-line-number="55"><span class="co">#&gt;  pkgload       1.0.2      2018-10-29 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-56" data-line-number="56"><span class="co">#&gt;  plyr          1.8.4      2016-06-08 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-57" data-line-number="57"><span class="co">#&gt;  prettyunits   1.0.2      2015-07-13 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-58" data-line-number="58"><span class="co">#&gt;  processx      3.3.0      2019-03-10 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-59" data-line-number="59"><span class="co">#&gt;  ps            1.3.0      2018-12-21 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-60" data-line-number="60"><span class="co">#&gt;  purrr         0.3.1      2019-03-03 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-61" data-line-number="61"><span class="co">#&gt;  qtl2        * 0.19-6     2019-03-13 [1] Github (rqtl/qtl2@eeb6f6d)    </span></a>
<a class="sourceLine" id="cb26-62" data-line-number="62"><span class="co">#&gt;  qtl2pleio   * 0.1.2.9001 2019-03-13 [1] local                         </span></a>
<a class="sourceLine" id="cb26-63" data-line-number="63"><span class="co">#&gt;  R6            2.4.0      2019-02-14 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-64" data-line-number="64"><span class="co">#&gt;  Rcpp          1.0.0.3    2019-03-08 [1] Github (RcppCore/Rcpp@08344ae)</span></a>
<a class="sourceLine" id="cb26-65" data-line-number="65"><span class="co">#&gt;  remotes       2.0.2      2018-10-30 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-66" data-line-number="66"><span class="co">#&gt;  rlang         0.3.1      2019-01-08 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-67" data-line-number="67"><span class="co">#&gt;  rmarkdown     1.11       2018-12-08 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-68" data-line-number="68"><span class="co">#&gt;  roxygen2      6.1.1      2018-11-07 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-69" data-line-number="69"><span class="co">#&gt;  rprojroot     1.3-2      2018-01-03 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-70" data-line-number="70"><span class="co">#&gt;  RSQLite       2.1.1      2018-05-06 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-71" data-line-number="71"><span class="co">#&gt;  rstudioapi    0.9.0      2019-01-09 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-72" data-line-number="72"><span class="co">#&gt;  scales        1.0.0      2018-08-09 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-73" data-line-number="73"><span class="co">#&gt;  sessioninfo   1.1.1      2018-11-05 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-74" data-line-number="74"><span class="co">#&gt;  stringi       1.4.3      2019-03-12 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-75" data-line-number="75"><span class="co">#&gt;  stringr       1.4.0      2019-02-10 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-76" data-line-number="76"><span class="co">#&gt;  testthat      2.0.1      2018-10-13 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-77" data-line-number="77"><span class="co">#&gt;  tibble        2.0.1      2019-01-12 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-78" data-line-number="78"><span class="co">#&gt;  tidyselect    0.2.5      2018-10-11 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-79" data-line-number="79"><span class="co">#&gt;  usethis       1.4.0      2018-08-14 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-80" data-line-number="80"><span class="co">#&gt;  utf8          1.1.4      2018-05-24 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-81" data-line-number="81"><span class="co">#&gt;  withr         2.1.2      2018-03-15 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-82" data-line-number="82"><span class="co">#&gt;  xfun          0.5        2019-02-20 [1] CRAN (R 3.5.2)                </span></a>
<a class="sourceLine" id="cb26-83" data-line-number="83"><span class="co">#&gt;  xml2          1.2.0      2018-01-24 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-84" data-line-number="84"><span class="co">#&gt;  yaml          2.2.0      2018-07-25 [1] CRAN (R 3.5.0)                </span></a>
<a class="sourceLine" id="cb26-85" data-line-number="85"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb26-86" data-line-number="86"><span class="co">#&gt; [1] /Library/Frameworks/R.framework/Versions/3.5/Resources/library</span></a></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#setting-for-pleiotropy-vs-two-separate-qtl">Setting for pleiotropy vs. two separate QTL</a></li>
      <li><a href="#installing-qtl2pleio">Installing <code>qtl2pleio</code></a></li>
      <li><a href="#reading-data-from-qtl2data-repository-on-github">Reading data from <code>qtl2data</code> repository on github</a></li>
      <li><a href="#kinship-calculations">Kinship calculations</a></li>
      <li><a href="#statistical-model">Statistical model</a></li>
      <li><a href="#simulating-phenotypes-with-qtl2pleiosim1">Simulating phenotypes with <code><a href="--/reference/sim1-html">qtl2pleio::sim1</a></code></a></li>
      <li>
<a href="#perform-two-dimensional-scan-as-first-step-in-pleiotropy-v-separate-qtl-hypothesis-test">Perform two-dimensional scan as first step in pleiotropy v separate QTL hypothesis test</a><ul class="nav nav-pills nav-stacked">
<li><a href="#session-info">Session info</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Frederick Boehm.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9000.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
