<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Tidy the data frame outputted by scan_pvl for further analysis &amp; plotting — tidy_scan_pvl • qtl2pleio</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Tidy the data frame outputted by scan_pvl for further analysis &amp; plotting — tidy_scan_pvl" />

<meta property="og:description" content="Tidy the data frame outputted by scan_pvl for further analysis &amp;amp; plotting" />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">qtl2pleio</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/bootstrap-cluster.html">Using HTCondor for bootstrap analysis</a>
    </li>
    <li>
      <a href="../articles/recla-analysis.html">Recla Analysis</a>
    </li>
    <li>
      <a href="../articles/testing-pleiotropy-vs-separate-qtl.html">Testing pleiotropy v separate QTL in multiparental populations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/fboehm/qtl2pleio">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Tidy the data frame outputted by scan_pvl for further analysis &amp; plotting</h1>
    <small class="dont-index">Source: <a href='https://github.com/fboehm/qtl2pleio/blob/master/R/tidy_scan_pvl.R'><code>R/tidy_scan_pvl.R</code></a></small>
    <div class="hidden name"><code>tidy_scan_pvl.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Tidy the data frame outputted by scan_pvl for further analysis &amp; plotting</p>
    
    </div>

    <pre class="usage"><span class='fu'>tidy_scan_pvl</span>(<span class='no'>mytib</span>, <span class='no'>pmap</span>, <span class='kw'>phenames</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"trait1"</span>, <span class='st'>"trait2"</span>))</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>mytib</th>
      <td><p>outputted tibble from scan_pvl</p></td>
    </tr>
    <tr>
      <th>pmap</th>
      <td><p>physical map (in Mb) or genetic map (in cM) for exactly one chromosome</p></td>
    </tr>
    <tr>
      <th>phenames</th>
      <td><p>vector of length 2 containing two trait names</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Other profile log-likelihood tibble functions: <code><a href='add_intercepts.html'>add_intercepts</a></code>,
  <code><a href='add_pmap.html'>add_pmap</a></code>,
  <code><a href='assemble_profile_tib.html'>assemble_profile_tib</a></code>,
  <code><a href='transform_loglik_mat.html'>transform_loglik_mat</a></code></p></div>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># read data</span>
<span class='no'>iron</span> <span class='kw'>&lt;-</span> <span class='kw pkg'>qtl2</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/qtl2/topics/read_cross2'>read_cross2</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/system.file'>system.file</a></span>(<span class='st'>"extdata"</span>, <span class='st'>"iron.zip"</span>, <span class='kw'>package</span><span class='kw'>=</span><span class='st'>"qtl2"</span>))
<span class='co'># insert pseudomarkers into map</span>
<span class='no'>map</span> <span class='kw'>&lt;-</span> <span class='kw pkg'>qtl2</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/qtl2/topics/insert_pseudomarkers'>insert_pseudomarkers</a></span>(<span class='no'>iron</span>$<span class='no'>gmap</span>, <span class='kw'>step</span><span class='kw'>=</span><span class='fl'>1</span>)
<span class='co'># calculate genotype probabilities</span>
<span class='no'>probs</span> <span class='kw'>&lt;-</span> <span class='kw pkg'>qtl2</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/qtl2/topics/calc_genoprob'>calc_genoprob</a></span>(<span class='no'>iron</span>, <span class='no'>map</span>, <span class='kw'>error_prob</span><span class='kw'>=</span><span class='fl'>0.002</span>)
<span class='co'># grab phenotypes and covariates; ensure that covariates have names attribute</span>
<span class='no'>pheno</span> <span class='kw'>&lt;-</span> <span class='no'>iron</span>$<span class='no'>pheno</span>
<span class='co'># leave-one-chromosome-out kinship matrices</span>
<span class='no'>kinship</span> <span class='kw'>&lt;-</span> <span class='kw pkg'>qtl2</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/qtl2/topics/calc_kinship'>calc_kinship</a></span>(<span class='no'>probs</span>, <span class='st'>"loco"</span>)$<span class='no'>`1`</span>
<span class='co'># get founder allele probabilites</span>
<span class='no'>probs</span> <span class='kw'>&lt;-</span> <span class='kw pkg'>qtl2</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/qtl2/topics/genoprob_to_alleleprob'>genoprob_to_alleleprob</a></span>(<span class='no'>probs</span>)$<span class='no'>`1`</span>
<span class='no'>ac</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/matrix'>matrix</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/numeric'>as.numeric</a></span>(<span class='no'>iron</span>$<span class='no'>covar</span>$<span class='no'>sex</span> <span class='kw'>==</span> <span class='st'>"m"</span>, <span class='kw'>ncol</span> <span class='kw'>=</span> <span class='fl'>1</span>))
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/colnames'>colnames</a></span>(<span class='no'>ac</span>) <span class='kw'>&lt;-</span> <span class='st'>"sex"</span>
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/colnames'>rownames</a></span>(<span class='no'>ac</span>) <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/colnames'>rownames</a></span>(<span class='no'>probs</span>)
<span class='no'>ss</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='scan_pvl.html'>scan_pvl</a></span>(<span class='kw'>probs</span> <span class='kw'>=</span> <span class='no'>probs</span>, <span class='kw'>pheno</span> <span class='kw'>=</span> <span class='no'>pheno</span>, <span class='kw'>kinship</span> <span class='kw'>=</span> <span class='no'>kinship</span>, <span class='kw'>addcovar</span> <span class='kw'>=</span> <span class='no'>ac</span>,
<span class='kw'>start_snp</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>n_snp</span> <span class='kw'>=</span> <span class='fl'>80</span>, <span class='kw'>n_cores</span> <span class='kw'>=</span> <span class='fl'>1</span>)</div><div class='output co'>#&gt; <span class='message'>starting covariance matrices estimation with data from 284 subjects.</span></div><div class='output co'>#&gt; <span class='message'>covariance matrices estimation completed.</span></div><div class='input'><span class='fu'>tidy_scan_pvl</span>(<span class='no'>ss</span>, <span class='kw'>pm</span> <span class='kw'>=</span> <span class='no'>map</span>)</div><div class='output co'>#&gt; <span style='color: #555555;'># A tibble: 240 x 4</span><span>
#&gt;    marker_position     lod trace      Trace     
#&gt;    </span><span style='color: #555555;font-style: italic;'>&lt;list&gt;</span><span>            </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>     
#&gt; </span><span style='color: #555555;'> 1</span><span> </span><span style='color: #555555;'>&lt;NULL&gt;</span><span>           0      pleiotropy pleiotropy
#&gt; </span><span style='color: #555555;'> 2</span><span> </span><span style='color: #555555;'>&lt;NULL&gt;</span><span>          -</span><span style='color: #BB0000;'>0.032</span><span style='color: #BB0000;text-decoration: underline;'>6</span><span> pleiotropy pleiotropy
#&gt; </span><span style='color: #555555;'> 3</span><span> </span><span style='color: #555555;'>&lt;NULL&gt;</span><span>          -</span><span style='color: #BB0000;'>0.068</span><span style='color: #BB0000;text-decoration: underline;'>1</span><span> pleiotropy pleiotropy
#&gt; </span><span style='color: #555555;'> 4</span><span> </span><span style='color: #555555;'>&lt;NULL&gt;</span><span>          -</span><span style='color: #BB0000;'>0.106</span><span>  pleiotropy pleiotropy
#&gt; </span><span style='color: #555555;'> 5</span><span> </span><span style='color: #555555;'>&lt;NULL&gt;</span><span>          -</span><span style='color: #BB0000;'>0.148</span><span>  pleiotropy pleiotropy
#&gt; </span><span style='color: #555555;'> 6</span><span> </span><span style='color: #555555;'>&lt;NULL&gt;</span><span>          -</span><span style='color: #BB0000;'>0.191</span><span>  pleiotropy pleiotropy
#&gt; </span><span style='color: #555555;'> 7</span><span> </span><span style='color: #555555;'>&lt;NULL&gt;</span><span>          -</span><span style='color: #BB0000;'>0.237</span><span>  pleiotropy pleiotropy
#&gt; </span><span style='color: #555555;'> 8</span><span> </span><span style='color: #555555;'>&lt;NULL&gt;</span><span>          -</span><span style='color: #BB0000;'>0.286</span><span>  pleiotropy pleiotropy
#&gt; </span><span style='color: #555555;'> 9</span><span> </span><span style='color: #555555;'>&lt;NULL&gt;</span><span>          -</span><span style='color: #BB0000;'>0.336</span><span>  pleiotropy pleiotropy
#&gt; </span><span style='color: #555555;'>10</span><span> </span><span style='color: #555555;'>&lt;NULL&gt;</span><span>          -</span><span style='color: #BB0000;'>0.388</span><span>  pleiotropy pleiotropy
#&gt; </span><span style='color: #555555;'># … with 230 more rows</span><span></div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Frederick J Boehm.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9000.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

